---
title: Multiomic responses to seasonal and diurnal cycles in
  Ostreococcus
author: "Francisco J. Romero-Campero & Ana B. Romero-Losada"
date: "August 11, 2020"
output:
  pdf_document: default
  html_document:
    df_print: paged
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## **Experimental design**

In this project we study the transcriptomic, proteomic and physiological responses to seasonal
and diurnal cycles in the picoeukaryote *Ostreococcus tauri*. Our favourite microalgae
was grown in 1.8L column photochemostats under long day conditions (16 hours light : 8 hours dark) representing a summer day and under short day conditions (8 hours light : 16 hours dark) simulating a winter day. After four weeks of entrainment under each condition samples were collected for three days every four hours. Then the program controlling the lights in the photochemostats was set to free running conditions consisting on constant light (LL) in half the systems and constant darkness (DD) in the other half. After a day, samples were again collected every four hours for two days under LL and DD. 

## **Load data and principal component analysis.**

The matrix containing the gene expression data analyzed in this study can be
downloaded from [**this link from the GEO data base**](https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE155535&format=file&file=GSE155535%5Fgene%5Fexpression%2Etsv%2Egz). Make sure to uncompress this file and rename it to **gene_expression.tsv**. First, the gene expression data is loaded and converted into a 
numeric matrix setting the rownames to gene ids. We also load gene annotation.

```{r load_data}
gene.expression <- read.table(file ="gene_expression.tsv",sep="\t",header=T,as.is=T)
head(gene.expression[,1:7])

gene.ids <- gene.expression$X

gene.expression <- as.matrix(gene.expression[,2:ncol(gene.expression)])
rownames(gene.expression) <- gene.ids
head(gene.expression[,1:6])

gene.annotation <- read.table(file="annotation/ostreococcus_tauri_annotation.tsv",header = T, as.is=T,sep="\t",quote = "")
head(gene.annotation)
```

The current version of *Ostreococcus tauri* genome available from [**here**](https://mycocosm.jgi.doe.gov/Ostta4221_3/Ostta4221_3.home.html) identifies 7668 genes. In our experiment only 3 genes were never expressed and 40, 164 and 260 genes never presented an expression level greater than 1, 5 and 10 FPKM respectively. This shows that practically the entire transcriptome of Ostreococcus is expressed under the seasonal and diurnal cycles studied in this project.

```{r number_genes}
ostta.genes <- rownames(gene.expression)
number.genes <- length(ostta.genes)
number.genes

dim(gene.expression)

length(which(apply(X = gene.expression,MARGIN = 1,FUN = max) == 0))
length(which(apply(X = gene.expression,MARGIN = 1,FUN = max) < 1))
length(which(apply(X = gene.expression,MARGIN = 1,FUN = max) < 5))
length(which(apply(X = gene.expression,MARGIN = 1,FUN = max) < 10))
```


We focus on the data generated under long and short days cycles by extracting them from the gene expression matrix. The resulting matrix has 7668 rows representing genes and 36 columns. This number of columns correspond to 36 different data points, each day is represented by 6 data points and we took samples for three days under long day conditions, and short day conditions.


```{r extract_sd_ld}
ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
ld.zt.i <- sapply(X = ld.zt,FUN = function(x){ paste(x,1:3,sep="_")})
sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
sd.zt.i <- sapply(X = sd.zt,FUN = function(x){ paste(x,1:3,sep="_")})

ld.sd.gene.expression <- gene.expression[,c(ld.zt.i,sd.zt.i)]
head(ld.sd.gene.expression[,1:6])
dim(ld.sd.gene.expression)
```

We perform **Principal Component Analysis** and a **Hierarchical clustering** in order to uncover the underlying structure in our data. We use the packages FactoMineR and factoextra and reformat the data as needed for the function PCA.

```{r pca_ld_sd}
library(FactoMineR)
library(factoextra)

pca.gene.expression.ld.sd <- data.frame(colnames(ld.sd.gene.expression),t(ld.sd.gene.expression))
colnames(pca.gene.expression.ld.sd)[1] <- "Time point"

res.pca.ld.sd <- PCA(pca.gene.expression.ld.sd, graph = FALSE,scale.unit = TRUE,quali.sup = 1 )
res.hcpc.ld.sd <- HCPC(res.pca.ld.sd, graph=FALSE)   
fviz_dend(res.hcpc.ld.sd,k=3,
          cex = 1,                       # Label size
          palette = "jco",               # Color palette see ?ggpubr::ggpar
          rect = TRUE, rect_fill = TRUE, # Add rectangle around groups
          rect_border = "jco",           # Rectangle color
          type="rectangle",
          labels_track_height = 400      # Augment the room for labels
)
```


The transcriptomes corresponding to the same time during the three different days under both LD and SD conditions tend to cluster together. This indicates a high circadian synchronization in our cultures. Using hierarchical clustering, the 36 transcriptomes under LD and SD conditions assemble together into three different groups. The first cluster corresponds to **midday**. The transcriptomes at time points ZT4 and ZT8 under LD and ZT4 under SD constitute this cluster. These time points correspond to the moments of maximal incident light irradiance under both LD and SD conditions. The second cluster conforms the **dusk** group. Here the transcriptomes at time points ZT12 and ZT16 under LD and ZT8 under SD are found. These time points coincide with the end of the light period in both LD and SD conditions when light irradiance is low. The third cluster represents **night/dawn** and comprises the transcriptomes at time points ZT20, ZT0 under LD and ZT12, ZT16, ZT20 and ZT0 under SD. The transcriptomes at time points in the LD and SD nights or dark periods constitute two distinct groups suggesting noticeable differences in the transcriptomic responses during the night under LD and SD conditions. It is also noteworthy the higher similarity between the dusk, night/dawn transcriptomes when compare to the midday ones. 

In order to obtain a more precise understanding of the underlying structure in our data we preform PCA separately for the LD and SD data. 

```{r pca_ld}
ld.gene.expression <- gene.expression[,ld.zt.i]
pca.gene.expression.ld <- data.frame(colnames(ld.gene.expression),t(ld.gene.expression))
colnames(pca.gene.expression.ld)[1] <- "Time point"

res.pca.ld <- PCA(pca.gene.expression.ld, graph = FALSE,scale.unit = TRUE,quali.sup = 1 )
fviz_pca_ind(res.pca.ld, col.ind = c("ZT00", "ZT00", "ZT00", "ZT04", "ZT04", "ZT04",
                                     "ZT08", "ZT08", "ZT08", "ZT12", "ZT12", "ZT12",
                                     "ZT16", "ZT16", "ZT16", "ZT20", "ZT20", "ZT20"), 
             pointsize=2, pointshape=21,fill="black",
             repel = TRUE, 
             addEllipses = TRUE,ellipse.type = "confidence",
             legend.title="Conditions",
             title="",
             show_legend=TRUE,show_guide=TRUE) 
```

Under LD conditions, we observed that the transcriptomes corresponding to the same time point in the three different days tightly cluster together globally constituting a circular structure. Next, we analyze the SD data.


```{r pca_sd}
sd.gene.expression <- gene.expression[,sd.zt.i]
pca.gene.expression.sd <- data.frame(colnames(sd.gene.expression),t(sd.gene.expression))
colnames(pca.gene.expression.sd)[1] <- "Time point"

res.pca.sd <- PCA(pca.gene.expression.sd, graph = FALSE,scale.unit = TRUE,quali.sup = 1 )
fviz_pca_ind(res.pca.sd, col.ind = c("ZT00", "ZT00", "ZT00", "ZT04", "ZT04", "ZT04",
                                     "ZT08", "ZT08", "ZT08", "ZT12", "ZT12", "ZT12",
                                     "ZT16", "ZT16", "ZT16", "ZT20", "ZT20", "ZT20"), 
             pointsize=2, pointshape=21,fill="black",
             repel = TRUE, 
             addEllipses = TRUE,ellipse.type = "confidence",
             legend.title="Conditions",
             title="",
             show_legend=TRUE,show_guide=TRUE) 
```

Under SD conditions more variability is observed and the time point transcriptomes form a structure resembling an ellipse. This could indicate that whereas in LD conditions gene expression in globally cycling precisely with a similar period a more complex behavior is expected under SD conditions. Also, it is remarkable the high similarity between the transcriptomes corresponding to ZT0 and ZT20 under SD conditions that is not present under LD conditions. This suggests that the transcriptomic response at the end of a SD night is already preparing all molecular systems for the incoming light availability at dawn whereas this anticipation is not as clearly observed under LD conditions. In overall, these results support that our experimental design grants a high level of synchronization in our data allowing us to proceed to the identification and comparison of genes exhibiting rhythmic expression patterns under LD and SD conditions. 

In order to get a more complete representation of different photoperiod and seasonality in our study we included previously published microarrays data generated under neutral day (ND) conditions 12h light : 12h dark. We performed PCA analysis to confirm synchronization in these data. Similar to the LD conditions the transcriptomes corresponding to the same data points from different days cluster together emerging a circular structure unveiling rhythmic gene expression.   

```{r pca_nd}
nd.gene.expression <- as.matrix(read.table(file ="nd_gene_expression.tsv",header=T,sep="\t",as.is=T))
head(nd.gene.expression[,1:7])

pca.gene.expression.nd <- data.frame(colnames(nd.gene.expression),t(nd.gene.expression))
colnames(pca.gene.expression.nd)[1] <- "Time point"

res.pca.nd <- PCA(pca.gene.expression.nd, graph = FALSE,scale.unit = TRUE,quali.sup = 1 )
fviz_pca_ind(res.pca.nd, col.ind = c("ZT00", "ZT00", "ZT00", "ZT03", "ZT03", "ZT03",
                                     "ZT06", "ZT06", "ZT06", "ZT09", "ZT09", "ZT09",
                                     "ZT12", "ZT12", "ZT12", "ZT15", "ZT15", "ZT15", 
                                     "ZT18", "ZT18", "ZT18", "ZT21", "ZT21", "ZT21"), 
             pointsize=2, pointshape=21,fill="black",
             repel = TRUE, 
             addEllipses = TRUE,ellipse.type = "confidence",
             legend.title="Conditions",
             title="",
             show_legend=TRUE,show_guide=TRUE) 


```

## **Identification of genes exhibiting rhythmic expression patterns under different photoperiods**

Using the bioconductor library **RAIN (Rhythmicity Analysis Incorporating Nonparametric methods)** we identified genes exhibiting periodic or rhythmic expression patterns under the different photoperiods. A p-value of 0.05 was used in all the analysis. Simple rhythmic expression patterns consisting of a single expression peak were identified by searching for patterns with a period of 24 hours. More complex rhythmic gene expression patterns such as genes presenting two or three peaks in a single day were searched by fixing periods of 12 and 8 hours. 

```{r ld_rain}
library(rain)

results.ld <- rain(t(ld.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=3)
sum(results.ld$pVal < 0.05)/number.genes
rhythmic.genes.ld <- rownames(subset(results.ld, pVal < 0.05))
length(rhythmic.genes.ld)

results.ld.12 <- rain(t(ld.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=3)
sum(results.ld.12$pVal < 0.05)/number.genes
rhythmic.genes.ld.12 <- rownames(subset(results.ld.12, pVal < 0.05))
length(rhythmic.genes.ld.12)

complete.ld.rhythmic.genes <- unique(c(rhythmic.genes.ld,rhythmic.genes.ld.12))
length(complete.ld.rhythmic.genes)/number.genes

non.rhythmic.ld.genes <- setdiff(ostta.genes,complete.ld.rhythmic.genes)
length(non.rhythmic.ld.genes)
length(non.rhythmic.ld.genes)/number.genes

length(setdiff(rhythmic.genes.ld,rhythmic.genes.ld.12))
length(setdiff(rhythmic.genes.ld,rhythmic.genes.ld.12))/number.genes

results.ld.8 <- rain(t(ld.gene.expression), deltat=4, period=8, verbose=FALSE, nr.series=3)
sum(results.ld.8$pVal < 0.05)/number.genes

write.table(x = subset(gene.annotation, geneID %in% non.rhythmic.ld.genes),file="gene_sets/LD_non_rhythmic_genes.tsv",row.names = F)

write.table(x = subset(gene.annotation, geneID %in% rhythmic.genes.ld.12),file="gene_sets/LD_rhythmic_genes_2_peaks.tsv",row.names = F)
```

Under LD conditions we found that 6201 genes constituting 80.87% of the *Ostreococcus tauri* genome exhibit diurnal rhythmic expression patterns. Most of these genes, 5825 genes covering 75.97% of the entire genome presented a simple rhythmic expression pattern with a 24 hours period and a single expression peak. Only 376 genes presented a more complex rhythmic expression pattern presenting two expression peaks with an apparent period of 12 hours in a single day. The number of genes with expression patterns presenting three peaks in a single day with an apparent period of 8 hours was negligible, 81 genes representing less than 1% of the genome. We also identified 1467 genes comprising 19.13% of genome that did not present a detectable rhythmic expression pattern.      

```{r nd_rain}
results.nd <- rain(t(nd.gene.expression), deltat=3, period=24, verbose=FALSE, nr.series=3)
sum(results.nd$pVal < 0.05)/number.genes
rhythmic.genes.nd <- intersect(rownames(subset(results.nd, pVal < 0.05)),ostta.genes)
length(rhythmic.genes.nd)

results.nd.12 <- rain(t(nd.gene.expression), deltat=3, period=12, verbose=FALSE, nr.series=3)
sum(results.nd.12$pVal < 0.05)/number.genes
rhythmic.genes.nd.12 <- rownames(subset(results.nd.12, pVal < 0.05))
length(rhythmic.genes.nd.12)

complete.nd.rhythmic.genes <- unique(c(rhythmic.genes.nd,rhythmic.genes.nd.12))
length(complete.nd.rhythmic.genes)
length(complete.nd.rhythmic.genes)/number.genes

non.rhythmic.nd.genes <- setdiff(ostta.genes,complete.nd.rhythmic.genes)
length(non.rhythmic.nd.genes)

length(setdiff(rhythmic.genes.nd,rhythmic.genes.nd.12))
length(setdiff(rhythmic.genes.nd,rhythmic.genes.nd.12))/number.genes

results.nd.8 <- rain(t(nd.gene.expression), deltat=4, period=8, verbose=FALSE, nr.series=3)
sum(results.nd.8$pVal < 0.05)/number.genes ## 0.8%

write.table(x = subset(gene.annotation, geneID %in% non.rhythmic.nd.genes),file="gene_sets/ND_non_rhythmic_genes.tsv",row.names = F)

write.table(x = subset(gene.annotation, geneID %in% rhythmic.genes.nd.12),file="gene_sets/ND_rhythmic_genes_2_peaks.tsv",row.names = F)
```

Under ND conditions we found that 6372 genes constituting 83.1% of the *Ostreococcus tauri* genome exhibit diurnal rhythmic expression patterns. Most of these genes, 5201 genes covering 67.82%% of the entire genome presented a simple rhythmic expression pattern with a 24 hours period and a single expression peak. An increasing number of genes, namely 1171 genes, presented a more complex rhythmic expression pattern cycling twice in a day presenting two expression peaks with an apparent period of 12 hours. The number of genes with expression patterns cycling three times in a day with an apparent period of 8 hours was negligible, 61 genes representing less than 1% of the genome. We also identified 1274 genes comprising 16.61% of genome that did not present a detectable rhythmic expression pattern.

```{r sd_rain}
results.sd <- rain(t(sd.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=3)
sum(results.sd$pVal < 0.05)/number.genes
rhythmic.genes.sd <- rownames(subset(results.sd, pVal < 0.05))
length(rhythmic.genes.sd)

results.sd.12 <- rain(t(sd.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=3)
sum(results.sd.12$pVal < 0.05)/number.genes
rhythmic.genes.sd.12 <- rownames(subset(results.sd.12, pVal < 0.05))
length(rhythmic.genes.sd.12)

complete.sd.rhythmic.genes <- unique(c(rhythmic.genes.sd,rhythmic.genes.sd.12))
length(complete.sd.rhythmic.genes)
length(complete.sd.rhythmic.genes)/number.genes

non.rhythmic.sd.genes <- setdiff(ostta.genes,complete.sd.rhythmic.genes)
length(non.rhythmic.sd.genes)
length(non.rhythmic.sd.genes)/number.genes

results.sd.8 <- rain(t(sd.gene.expression), deltat=4, period=8, verbose=FALSE, nr.series=3)
sum(results.sd.8$pVal < 0.05)
sum(results.sd.8$pVal < 0.05)/number.genes

write.table(x = subset(gene.annotation, geneID %in% non.rhythmic.sd.genes),file="gene_sets/SD_non_rhythmic_genes.tsv",row.names = F)

write.table(x = subset(gene.annotation, geneID %in% rhythmic.genes.sd.12),file="gene_sets/SD_rhythmic_genes_2_peaks.tsv",row.names = F)
```

Under SD conditions we found that 6104 genes constituting 79.6% of the *Ostreococcus tauri* genome exhibit diurnal cycling expression patterns. A substantially reduced number of these genes, 4249 genes covering 55.41%% of the entire genome presented a simple rhythmic expression pattern with a 24 hours period and a single expression peak. An increasing number of genes, namely 1855 genes, presented a more complex rhythmic expression pattern cycling twice in a day presenting two expression peaks with an apparent period of 12 hours. The number of genes with expression patterns cycling three times in a day with an apparent period of 8 hours was negligible, 23 genes representing less than 1% of the genome. We also identified 1564 genes comprising 20.41% of genome that did not present a detectable rhythmic expression pattern.


```{r creinhardtii_two_peaks}
expression.level <- read.table(file = "creinhardtii_expression_level.txt",header=T,as.is=T)

cre.expression <- as.matrix(expression.level[,2:ncol(expression.level)])
rownames(cre.expression) <- expression.level$gene

results.cre <- rain(t(cre.expression), deltat=3, period=12, verbose=FALSE, nr.series=2)
sum(results.cre$pVal < 0.05)/nrow(cre.expression)
rhythmic.genes.cre.12 <- rownames(subset(results.cre, pVal < 0.05))
length(rhythmic.genes.cre.12)
```

The existence of genes exhibiting complex rhythmic expression patterns consisting of two peaks over a complete day is present in the transcriptome of other microalgae not being restricted only to Ostreococcus. We reanalyzed previously published RNA-seq data generated in Chlamydomonas under ND conditions and found that 2190 genes comprising 14.02% of the entire genome exhibit two expression peaks per day with an apparent period of 12 hours.

```{r rhythmic_barplot}
data <- matrix(c(length(rhythmic.genes.ld.12),length(setdiff(rhythmic.genes.ld,rhythmic.genes.ld.12)),length(non.rhythmic.ld.genes),
length(rhythmic.genes.nd.12),length(setdiff(rhythmic.genes.nd,rhythmic.genes.nd.12)),length(non.rhythmic.nd.genes),
length(rhythmic.genes.sd.12),length(setdiff(rhythmic.genes.sd,rhythmic.genes.sd.12)),length(non.rhythmic.sd.genes)),nrow=3)

barplot(data, lwd=2, names.arg = c("LD","ND","SD"),cex.names = 2,cex.axis = 1.5,
        col=colors()[c(23,89,12)], 
        border="black", 
        space=0.25, 
        font.axis=2, 
        xlab="",ylim=c(0,8000))
```

We observed that practically the same number of genes are identified as rhythmic under the three different conditions. Independently from the length of the photoperiod we identified approximately 80% of the genome of *Ostreococcus tauri* as rhythmically expressed. Remarkably, the number of rhythmic genes exhibiting two peaks during a day increases as the photoperiod shortens whereas the number of genes with a single peak in a day decreases. Furthermore, the number of genes that did not exhibit a diurnal rhythmic pattern was similar under the three different photoperiods. 


```{r ld_nd_sd_rhythmic_genes}
library(VennDiagram)
grid.newpage()
draw.triple.venn(area1 = length(complete.ld.rhythmic.genes),area2 = length(complete.sd.rhythmic.genes),area3 = length(complete.nd.rhythmic.genes),n12 = length(intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes)),n23 = length(intersect(complete.sd.rhythmic.genes,complete.nd.rhythmic.genes)), n13 = length(intersect(complete.ld.rhythmic.genes,complete.nd.rhythmic.genes)), n123 = length(intersect(intersect(complete.sd.rhythmic.genes, complete.ld.rhythmic.genes),complete.nd.rhythmic.genes)),lwd = 3,category = c("LD","SD","ND"),euler.d = T,col = c("blue","red","darkgrey"),fill = c(" blue","red","darkgrey"),alpha = 0.3,cex = 2,cat.cex = 2)

length(intersect(intersect(complete.sd.rhythmic.genes, complete.ld.rhythmic.genes),complete.nd.rhythmic.genes))

length(intersect(intersect(complete.sd.rhythmic.genes, complete.ld.rhythmic.genes),complete.nd.rhythmic.genes))/number.genes
```

When comparing rhythmically expressed genes under LD, ND and SD conditions we found that 4589 genes comprising 59.84% of the genome exhibit rhythmic expression patterns. Whereas only a few hundreds genes only presented a cycling expression pattern under a specific photoperiod. In order to further explore the differences between rhythmic a non rhythmic genes we compared their expression level under LD and SD conditions. We restrict our analysis to these two conditions since here expression levels are estimated using the same technology namely RNA-seq whereas for ND microarrays were used. 

```{r non_rhythmic_expression}

rhythmic.max.expression <- apply(X = gene.expression[intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes),c(sd.zt.i,ld.zt.i)],MARGIN = 1,FUN = max)
non.rhythmic.max.expression <- apply(X = gene.expression[intersect(non.rhythmic.ld.genes,non.rhythmic.sd.genes),c(sd.zt.i,ld.zt.i)], MARGIN = 1,FUN = max)
boxplot(rhythmic.max.expression, non.rhythmic.max.expression, outline=F,col=colors()[c(89,12)],names=c("Rhythmic","Non rhythmic"),ylab="FPKM",cex.lab=1.5,las=2)

wilcox.test(rhythmic.max.expression,3*non.rhythmic.max.expression,alternative="greater")
```

We found that genes exhibiting rhythmic expression patterns under both LD and SD conditions present a maximal level of expression three times greater than genes identified as non rhythmic. This difference was significant according to p-value of 1.45e-4 computed using Mann-Whitney-Wilcoxson test. The current methods for detecting rhythmic gene expression are known to only perform optimally for highly expressed genes. Therefore, the genes identified as non rhythmic in this study could indeed be rhythmic although their low expression level have prevented our methods to detect them.

## **Identification of bona fide circadian genes and light/dark regulated genes under different photoperiods**

Genes maintaining their rhythmic expression pattern in continuous light regimes (LL conditions) independently from the alternating light and dark periods are considered here bona fide circadian genes. Our experimental design includes two days under continuous light representing free running conditions after the three days under either LD or SD conditions. We use RAIN to identify circadian genes. 

```{r ld_ll_rain}
ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
#ld.ll.zt.i <- sapply(X = ld.zt,FUN = function(x){ paste(x,2:5,sep="_")})
ld.ll.zt.i <- sapply(X = ld.zt,FUN = function(x){ paste(x,4:5,sep="_")})

ld.ll.gene.expression <- gene.expression[,ld.ll.zt.i]
dim(ld.ll.gene.expression)

#results.ld.ll <- rain(t(ld.ll.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=4)
results.ld.ll <- rain(t(ld.ll.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=2)
head(results.ld.ll)
sum(results.ld.ll$pVal < 0.05)/number.genes 
rhythmic.genes.ld.ll <- rownames(subset(results.ld.ll, pVal < 0.05))
length(rhythmic.genes.ld.ll)
rhythmic.genes.ld.ll <- intersect(complete.ld.rhythmic.genes,rhythmic.genes.ld.ll)
length(rhythmic.genes.ld.ll)/number.genes 

#results.ld.ll.12 <- rain(t(ld.ll.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=4)
results.ld.ll.12 <- rain(t(ld.ll.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=2)
sum(results.ld.ll.12$pVal < 0.05)/number.genes
rhythmic.genes.ld.ll.12 <- rownames(subset(results.ld.ll.12, pVal < 0.05))
length(rhythmic.genes.ld.ll.12)
rhythmic.genes.ld.ll.12 <- intersect(rhythmic.genes.ld.ll.12,complete.ld.rhythmic.genes)
complete.ld.ll.rhythmic.genes <- unique(c(rhythmic.genes.ld.ll,rhythmic.genes.ld.ll.12))
length(complete.ld.ll.rhythmic.genes)
length(complete.ld.ll.rhythmic.genes)/length(complete.ld.rhythmic.genes)
length(complete.ld.ll.rhythmic.genes)/number.genes

length(setdiff(complete.ld.rhythmic.genes, complete.ld.ll.rhythmic.genes))

length(intersect(rhythmic.genes.ld.ll.12,rhythmic.genes.ld.12))
length(intersect(rhythmic.genes.ld.ll,rhythmic.genes.ld.12))

genes.two.peaks.ld.one.peak.ll <- intersect(rhythmic.genes.ld.ll,rhythmic.genes.ld.12)

length(rhythmic.genes.ld.ll.12)

length(rhythmic.genes.ld.12)

non.rhythmic.ld.ll.genes <- setdiff(complete.ld.rhythmic.genes, complete.ld.ll.rhythmic.genes)
length(complete.ld.ll.rhythmic.genes)/number.genes
length(complete.ld.ll.rhythmic.genes)/length(complete.ld.rhythmic.genes)

length(setdiff(complete.ld.rhythmic.genes,complete.ld.ll.rhythmic.genes))

# data.ld.ll <- matrix(c(length(complete.ld.ll.rhythmic.genes),length(setdiff(complete.ld.rhythmic.genes,complete.ld.ll.rhythmic.genes)),length(non.rhythmic.ld.genes)),nrow=3)
# 
# sum(c(length(complete.ld.ll.rhythmic.genes),length(setdiff(complete.ld.rhythmic.genes,complete.ld.ll.rhythmic.genes)),length(non.rhythmic.ld.genes)))
# 
# barplot(data.ld.ll, lwd=2, names.arg = "",cex.names = 2,cex.axis = 1.5,
#         col=c("blue","lightblue","white"), 
#         border="black", 
#         font.axis=2, 
#         xlab="",ylim=c(0,8000))


#ll.zt.i <- sapply(X = ld.zt,FUN = function(x){ paste(x,4:5,sep="_")})
#max.expression.non.rhythmic.ld.ll <- apply(X = gene.expression[non.rhythmic.ld.ll.genes,ll.zt.i],MARGIN = 1,FUN = max)

#length(max.expression.non.rhythmic.ld.ll)
#sum(max.expression.non.rhythmic.ld.ll < 5)
#sum(max.expression.non.rhythmic.ld.ll < 10)
#sum(max.expression.non.rhythmic.ld.ll > 50)

#var.expression.non.rhythmic.ld.ll <- apply(X = gene.expression[non.rhythmic.ld.ll.genes,ll.zt.i],MARGIN = 1,FUN = var)

#sum(var.expression.non.rhythmic.ld.ll < 5)

write.table(x = subset(gene.annotation, geneID %in% complete.ld.ll.rhythmic.genes),file="gene_sets/LD_LL_rhythmic_genes.tsv",row.names = F)
```

We found that 45.22% of the genes identified previously as rhythmic under LD conditions still presented cycling expression patterns when transferred to free running conditions consisting of constant light (LL). Specifically, we identified 2804 genes comprising 36.57% of the entire genome that maintain rhythmicity under constant light after entraiment in LD conditions. The 3397 remaining genes cycling under LD conditions presented either a flat or noisy expression profile under LL conditions.      


```{r ld_ll_amplitude_mesor_phase}
library(circacompare)
ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")

circacompare.ld.ll <- matrix(nrow=length(complete.ld.ll.rhythmic.genes),ncol=15)
rownames(circacompare.ld.ll) <- complete.ld.ll.rhythmic.genes

for(i in 1:length(complete.ld.ll.rhythmic.genes))
{
  gene.i <- complete.ld.ll.rhythmic.genes[i]
  
  ld.expression.i <- gene.expression[gene.i,c(paste(ld.zt,2,sep="_"),paste(ld.zt,3,sep="_"))]
  ll.expression.i <- gene.expression[gene.i,c(paste(ld.zt,4,sep="_"),paste(ld.zt,5,sep="_"))]
  
  time.points <- seq(from=0,by=4,length.out = 12)
  
  ld.ll.df <- data.frame(time=c(time.points,time.points),
                         measure=c(ld.expression.i, ll.expression.i),
                         group=c(rep("ld",12),rep("ll",12)))
  
  out.i <- circacompare(x = ld.ll.df, col_time = "time", col_group = "group", col_outcome = "measure",alpha_threshold = 1)
  circacompare.ld.ll[i,] <- out.i[[2]][,2]
 }

colnames(circacompare.ld.ll) <- out.i[[2]][,1]

par(lwd=3)
boxplot(circacompare.ld.ll[,"ld amplitude estimate" ],circacompare.ld.ll[,"ll amplitude estimate" ],outline=F,col=c("blue","lightblue"),names=c("LD","LL"),cex.axis=2)
sum(circacompare.ld.ll[,"Amplitude difference estimate" ] < 0)
sum(circacompare.ld.ll[,"Amplitude difference estimate" ] < 0) / length(complete.ld.ll.rhythmic.genes)
sum(circacompare.ld.ll[,"P-value for amplitude difference" ] < 0.05)
sum(circacompare.ld.ll[,"P-value for amplitude difference" ] < 0.05)/length(complete.ld.ll.rhythmic.genes)

boxplot(circacompare.ld.ll[,"ld mesor estimate" ],circacompare.ld.ll[,"ll mesor estimate" ],outline=F,col=c("blue","lightblue"),names=c("LD","LL"),cex.axis=2)
sum(circacompare.ld.ll[,"Mesor difference estimate" ] < 0) /length(complete.ld.ll.rhythmic.genes)
sum(circacompare.ld.ll[,"Mesor difference estimate" ] > 0) /length(complete.ld.ll.rhythmic.genes)
sum(circacompare.ld.ll[,"P-value for mesor difference" ] < 0.05)/length(complete.ld.ll.rhythmic.genes)

sum(circacompare.ld.ll[,"Phase difference estimate"] > 0)
sum(circacompare.ld.ll[,"Phase difference estimate"] > 0)/length(complete.ld.ll.rhythmic.genes)
sum(circacompare.ld.ll[,"P-value for difference in phase" ] < 0.05)
sum(circacompare.ld.ll[,"P-value for difference in phase" ] < 0.05)/length(complete.ld.ll.rhythmic.genes)
```

Most circadian genes under LD conditions, 97.68%, presented a sharp decrease in amplitude when transferred to LL, being significant in more than half of them with p-value < 0.05. We also observed an increase in phase in 76.28% of the genes being significant in 14.98% of the genes exhibiting rhythmicity under constant light. Nevertheless, no clear effect was observed over the mesor that remained in LL similar to the values in LD. 


```{r ld_dd_rain}
ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
ld.dd.zt.i <- sapply(X = ld.zt,FUN = function(x){ paste(x,6:7,sep="_")})

ld.dd.gene.expression <- gene.expression[,ld.dd.zt.i]
dim(ld.dd.gene.expression)

#results.ld.dd <- rain(t(ld.dd.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=4)
results.ld.dd <- rain(t(ld.dd.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=2)
head(results.ld.dd)
sum(results.ld.dd$pVal < 0.05)/number.genes 
rhythmic.genes.ld.dd <- rownames(subset(results.ld.dd, pVal < 0.05))
length(rhythmic.genes.ld.dd)
rhythmic.genes.ld.dd <- intersect(complete.ld.rhythmic.genes,rhythmic.genes.ld.dd)
length(rhythmic.genes.ld.dd)/number.genes 

#results.ld.dd.12 <- rain(t(ld.dd.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=4)
results.ld.dd.12 <- rain(t(ld.dd.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=2)
sum(results.ld.dd.12$pVal < 0.05)/number.genes
rhythmic.genes.ld.dd.12 <- rownames(subset(results.ld.dd.12, pVal < 0.05))
length(rhythmic.genes.ld.dd.12)
rhythmic.genes.ld.dd.12 <- intersect(rhythmic.genes.ld.dd.12,complete.ld.rhythmic.genes)
length(rhythmic.genes.ld.dd.12)

complete.ld.dd.rhythmic.genes <- unique(c(rhythmic.genes.ld.dd,rhythmic.genes.ld.dd.12))

length(complete.ld.dd.rhythmic.genes)
length(setdiff(complete.ld.rhythmic.genes, complete.ld.dd.rhythmic.genes))
non.rhythmic.ld.dd.genes <- setdiff(complete.ld.rhythmic.genes, complete.ld.dd.rhythmic.genes)
length(complete.ld.dd.rhythmic.genes)/number.genes
length(complete.ld.dd.rhythmic.genes)/length(complete.ld.rhythmic.genes)

length(setdiff(complete.ld.rhythmic.genes,complete.ld.dd.rhythmic.genes))


length(intersect(rhythmic.genes.ld.dd.12,rhythmic.genes.ld.12))
length(intersect(rhythmic.genes.ld.dd,rhythmic.genes.ld.12))

genes.two.peaks.ld.one.peak.dd <- intersect(rhythmic.genes.ld.dd,rhythmic.genes.ld.12)

length(rhythmic.genes.ld.dd.12)
length(rhythmic.genes.ld.12)




#data.ld.dd <- matrix(c(length(complete.ld.dd.rhythmic.genes),length(setdiff(complete.ld.rhythmic.genes,complete.ld.dd.rhythmic.genes)),length(non.rhythmic.ld.genes)),nrow=3)

#sum(c(length(complete.ld.dd.rhythmic.genes),length(setdiff(complete.ld.rhythmic.genes,complete.ld.dd.rhythmic.genes)),length(non.rhythmic.ld.genes)))

#barplot(data.ld.dd, lwd=2, names.arg = "",cex.names = 2,cex.axis = 1.5,
#        col=c("blue","lightblue","white"), 
#        border="black", 
#        font.axis=2, 
#        xlab="",ylim=c(0,8000))


#rhythmic.ld.ll <- setdiff(complete.ld.ll.rhythmic.genes,complete.ld.dd.rhythmic.genes)
#length(rhythmic.ld.ll)

#rhythmic.ld.dd <- setdiff(complete.ld.dd.rhythmic.genes,complete.ld.ll.rhythmic.genes)
#length(rhythmic.ld.dd)

#rhythmic.ld <- setdiff(non.rhythmic.ld.dd.genes, complete.ld.ll.rhythmic.genes)
#length(rhythmic.ld)

#length(non.rhythmic.ld.genes)

write.table(x = subset(gene.annotation, geneID %in% complete.ld.dd.rhythmic.genes),file="gene_sets/LD_DD_rhythmic_genes.tsv",row.names = F)
```

We found that 53.39% of the genes identified previously as rhythmic under LD conditions still presented cycling expression patterns when transferred to free running conditions consisting of constant darkness. Specifically, we identified 3311 genes under DD conditions comprising 53.39% of the entire genome that maintain rhythmicity under constant darkness after entrainment in LD conditions. The 2890 remaining genes cycling under LD conditions presented either a flat or noisy expression profile under DD conditions.      

```{r ld_dd_amplitude_mesor_phase}
library(circacompare)
ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")

circacompare.ld.dd <- matrix(nrow=length(complete.ld.dd.rhythmic.genes),ncol=15)
rownames(circacompare.ld.dd) <- complete.ld.dd.rhythmic.genes
i <- 1
for(i in 1:length(complete.ld.dd.rhythmic.genes))
{
  gene.i <- complete.ld.dd.rhythmic.genes[i]
  
  ld.expression.i <- gene.expression[gene.i,c(paste(ld.zt,2,sep="_"),paste(ld.zt,3,sep="_"))]
  dd.expression.i <- gene.expression[gene.i,c(paste(ld.zt,6,sep="_"),paste(ld.zt,7,sep="_"))]
  
  time.points <- seq(from=0,by=4,length.out = 12)
  
  ld.dd.df <- data.frame(time=c(time.points,time.points),
                         measure=c(ld.expression.i, dd.expression.i),
                         group=c(rep("ld",12),rep("dd",12)))
  
  out.i <- circacompare(x = ld.dd.df, col_time = "time", col_group = "group", col_outcome = "measure",alpha_threshold = 1)
  circacompare.ld.dd[i,] <- out.i[[2]][,2]
 }

colnames(circacompare.ld.dd) <- out.i[[2]][,1]

par(lwd=3)
boxplot(circacompare.ld.dd[,"ld amplitude estimate" ],circacompare.ld.dd[,"dd amplitude estimate" ],outline=F,col=c("blue","darkblue"),names=c("LD","DD"),cex.axis=2)
sum(circacompare.ld.dd[,"Amplitude difference estimate" ] > 0)
sum(circacompare.ld.dd[,"Amplitude difference estimate" ] > 0) / length(complete.ld.dd.rhythmic.genes)
sum(circacompare.ld.dd[,"P-value for amplitude difference" ] < 0.05)
sum(circacompare.ld.dd[,"P-value for amplitude difference" ] < 0.05)/length(complete.ld.dd.rhythmic.genes)

boxplot(circacompare.ld.dd[,"ld mesor estimate" ],circacompare.ld.dd[,"dd mesor estimate" ],outline=F,col=c("blue","darkblue"),names=c("LD","DD"),cex.axis=2)
sum(circacompare.ld.dd[,"Mesor difference estimate" ] < 0) /length(complete.ld.dd.rhythmic.genes)
sum(circacompare.ld.dd[,"Mesor difference estimate" ] > 0) /length(complete.ld.dd.rhythmic.genes)
sum(circacompare.ld.dd[,"P-value for mesor difference" ] < 0.05)/length(complete.ld.ll.rhythmic.genes)

sum(circacompare.ld.dd[,"Phase difference estimate"] > 0)
sum(circacompare.ld.dd[,"Phase difference estimate"] > 0)/length(complete.ld.dd.rhythmic.genes)
sum(circacompare.ld.dd[,"P-value for difference in phase" ] < 0.05)
sum(circacompare.ld.dd[,"P-value for difference in phase" ] < 0.05)/length(complete.ld.dd.rhythmic.genes)
```

Most circadian genes under LD conditions, 79.43%, presented a sharp decrease in amplitude when transferred to DD, being significant in 31.62% of the genes exhibiting rythmicity under DD with p-value < 0.05. In contrast to the case under constant light we observed a decrease in phase under constant darkness. Specifically, 87.32% of the rhythmic genes under DD presented an anticipation in phase that was significant in 34.16% of them. No clear effect was observed over the mesor of gene expression profiles that remained in DD similar to the values in LD. 

```{r ld_rhytmic_ll_dd}
rhythmic.ld.ll.dd <- intersect(complete.ld.dd.rhythmic.genes,complete.ld.ll.rhythmic.genes)
length(rhythmic.ld.ll.dd)

length(rhythmic.ld.ll.dd)/number.genes

grid.newpage()
draw.pairwise.venn(area1 = length(complete.ld.ll.rhythmic.genes),
                   area2 = length(complete.ld.dd.rhythmic.genes),
                   cross.area = length(intersect(complete.ld.ll.rhythmic.genes,complete.ld.dd.rhythmic.genes)),
                   lwd = 6,alpha = 0.7,fill = c("lightblue","darkblue"),cex = 3,category = c("LD/LL","LD/DD"),cat.cex = 3,cat.pos = c(-10,10))

write.table(x = subset(gene.annotation, geneID %in% rhythmic.ld.ll.dd),file="gene_sets/LD_LL_DD_rhythmic_genes.tsv",row.names = F)

rhythmic.only.ll.or.dd <- setdiff(unique(c(complete.ld.dd.rhythmic.genes,complete.ld.ll.rhythmic.genes)),rhythmic.ld.ll.dd)
length(rhythmic.only.ll.or.dd)
length(rhythmic.only.ll.or.dd) / number.genes

write.table(x = subset(gene.annotation, geneID %in% rhythmic.only.ll.or.dd),file="gene_sets/LD_and_LL_or_DD_rhythmic_genes.tsv",row.names = F)

ld.rhythmic.only.ll <- setdiff(complete.ld.ll.rhythmic.genes,rhythmic.ld.ll.dd)
length(ld.rhythmic.only.ll)
write.table(x = ld.rhythmic.only.ll, file="gene_sets/LD_and_LL_rhythmic_genes.tsv",row.names = F)

ld.rhythmic.only.dd <- setdiff(complete.ld.dd.rhythmic.genes,rhythmic.ld.ll.dd)
length(ld.rhythmic.only.dd)
write.table(x = ld.rhythmic.only.dd, file="gene_sets/LD_and_DD_rhythmic_genes.tsv",row.names = F)
```

Integrating the results from the two free running conditions analysed in this study, namely, constant light and constant dark we identified 1647 genes whose rhythmicity is independent from the alternating light/dark cycles after LD entrainment. In contrast, 2821 genes (36.79%) only maintained its rhythmicity after LD entrainment either under constant light or dark. Specifically, 1664 genes exhibiting only rhythmicity under constant dark and 1157 genes under constant light. These genes need either an light or dark input to maintain their rhythmicity loosing it under the free running conditions excluding the corresponding necessary input. 

Next, we analyse the data under winter short day conditions.

```{r sd_ll_rain}
sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
sd.ll.zt.i <- sapply(X = sd.zt,FUN = function(x){ paste(x,4:5,sep="_")})

sd.ll.gene.expression <- gene.expression[,sd.ll.zt.i]

#results.sd.ll <- rain(t(sd.ll.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=4)
results.sd.ll <- rain(t(sd.ll.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=2)
head(results.sd.ll)
sum(results.sd.ll$pVal < 0.05)/number.genes ## 16.21%%
rhythmic.genes.sd.ll <- rownames(subset(results.sd.ll, pVal < 0.05))
rhythmic.genes.sd.ll <- intersect(rhythmic.genes.sd.ll, complete.sd.rhythmic.genes)
length(rhythmic.genes.sd.ll)

results.sd.ll.12 <- rain(t(sd.ll.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=2)
sum(results.sd.ll.12$pVal < 0.05)/number.genes ## 8.62%
rhythmic.genes.sd.ll.12 <- rownames(subset(results.sd.ll.12, pVal < 0.05))
rhythmic.genes.sd.ll.12 <- intersect(rhythmic.genes.sd.ll.12, complete.sd.rhythmic.genes)
length(rhythmic.genes.sd.ll.12)

length(intersect(rhythmic.genes.sd.ll.12,rhythmic.genes.sd.12))
length(intersect(rhythmic.genes.sd.ll.12,rhythmic.genes.sd.12)) / length(rhythmic.genes.sd.12) #10.78%

length(intersect(rhythmic.genes.sd.ll,rhythmic.genes.sd.12))
length(intersect(rhythmic.genes.sd.ll,rhythmic.genes.sd.12)) / length(rhythmic.genes.sd.12) #15.09%

genes.two.peaks.sd.one.peak.ll <- intersect(rhythmic.genes.sd.ll,rhythmic.genes.sd.12)

complete.sd.ll.rhythmic.genes <- unique(c(rhythmic.genes.sd.ll,rhythmic.genes.sd.ll.12))
length(complete.sd.ll.rhythmic.genes)
length(complete.sd.ll.rhythmic.genes)/length(complete.sd.rhythmic.genes)
length(complete.sd.ll.rhythmic.genes)/number.genes

length(setdiff(complete.sd.rhythmic.genes,complete.sd.ll.rhythmic.genes))
non.rhythmic.sd.ll.genes <- setdiff(complete.sd.rhythmic.genes, complete.sd.ll.rhythmic.genes)



data.sd.ll <- matrix(c(length(complete.sd.ll.rhythmic.genes),length(setdiff(complete.sd.rhythmic.genes,complete.sd.ll.rhythmic.genes)),length(non.rhythmic.sd.genes)),nrow=3)
sum(c(length(complete.sd.ll.rhythmic.genes),length(setdiff(complete.sd.rhythmic.genes,complete.sd.ll.rhythmic.genes)),length(non.rhythmic.sd.genes)))

barplot(data.sd.ll, lwd=2, names.arg = "",cex.names = 2,cex.axis = 1.5,
        col=c("red","lightpink","white"), 
        border="black", 
        font.axis=2, 
        xlab="",ylim=c(0,8000))

length(intersect(complete.ld.ll.rhythmic.genes,complete.sd.ll.rhythmic.genes))
length(setdiff(complete.ld.ll.rhythmic.genes,complete.sd.ll.rhythmic.genes))
length(setdiff(complete.sd.ll.rhythmic.genes,complete.ld.ll.rhythmic.genes))

complete.ld.sd.rhythmic.genes <- intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes)

grid.newpage()
draw.pairwise.venn(area1 = length(intersect(complete.ld.sd.rhythmic.genes,complete.ld.ll.rhythmic.genes)),area2 = length(intersect(complete.ld.sd.rhythmic.genes,complete.sd.ll.rhythmic.genes)),cross.area = length(intersect(complete.ld.sd.rhythmic.genes,intersect(complete.ld.ll.rhythmic.genes,complete.sd.ll.rhythmic.genes))),lwd = 3,category = c("LD","SD"),euler.d = T,col = c("blue","red"),fill = c(" blue","red"),alpha = 0.3,cex = 2,cat.cex = 2)


length(intersect(intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes),setdiff(complete.sd.ll.rhythmic.genes,complete.ld.ll.rhythmic.genes)))

length(intersect(intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes),setdiff(complete.ld.ll.rhythmic.genes,complete.sd.ll.rhythmic.genes)))

data.ld.ll.dd <- matrix(c(length(rhythmic.ld.ll.dd),
                          length(rhythmic.ld.ll),
                          length(rhythmic.ld.dd),
                          length(rhythmic.ld),
                          length(non.rhythmic.ld.genes)),nrow=5)

sum(data.ld.ll.dd)

barplot(data.ld.ll.dd, lwd=2, names.arg = "",cex.names = 2,cex.axis = 1.5,
        col=colorRampPalette(c("blue", "white"))(20)[c(1,10,14,17,20)], 
        border="black", 
        font.axis=2, 
        xlab="",ylim=c(0,8000))


```

Unlike the analysis for LD conditions, we found that only 25% of the genes identified previously as rhythmic under SD conditions still presented cycling expression patterns when transferred to free running conditions consisting of constant light. Specifically, we only identified 1526 circadian genes under SD conditions comprising 19.9% of the entire genome. When comparing circadian genes under LD and SD conditions we found that most circadian genes in SD conditions are also circadian under LD conditions. Nonetheless, 2129 circadian genes were specific under LD conditions. This suggests that rhythmic expression under SD conditions is very dependent on the presence of a long period of dark whereas rhythmic expression under LD conditions due to the presence of a short period of dark is better maintained under continuous light.        


```{r sd_dd_rain}
sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
sd.dd.zt.i <- sapply(X = sd.zt,FUN = function(x){ paste(x,c(6,7),sep="_")})

sd.dd.gene.expression <- gene.expression[,sd.dd.zt.i]

results.sd.dd <- rain(t(sd.dd.gene.expression), deltat=4, period=24, verbose=FALSE, nr.series=2)
head(results.sd.dd)
sum(results.sd.dd$pVal < 0.05)/number.genes ## 60.93%
rhythmic.genes.sd.dd <- rownames(subset(results.sd.dd, pVal < 0.05))
rhythmic.genes.sd.dd <- intersect(rhythmic.genes.sd.dd, complete.sd.rhythmic.genes)
length(rhythmic.genes.sd.dd)

results.sd.dd.12 <- rain(t(sd.dd.gene.expression), deltat=4, period=12, verbose=FALSE, nr.series=2)
sum(results.sd.dd.12$pVal < 0.05)/number.genes ## 1.79%
rhythmic.genes.sd.dd.12 <- rownames(subset(results.sd.dd.12, pVal < 0.05))
rhythmic.genes.sd.dd.12 <- intersect(rhythmic.genes.sd.dd.12, complete.sd.rhythmic.genes)
length(rhythmic.genes.sd.dd.12)

length(intersect(rhythmic.genes.sd.dd.12,rhythmic.genes.sd.12))
length(intersect(rhythmic.genes.sd.dd.12,rhythmic.genes.sd.12)) / length(rhythmic.genes.sd.12) #1.67%

length(intersect(rhythmic.genes.sd.dd,rhythmic.genes.sd.12))
length(intersect(rhythmic.genes.sd.dd,rhythmic.genes.sd.12)) / length(rhythmic.genes.sd.12) #74.39%

genes.two.peaks.sd.one.peak.dd <- intersect(rhythmic.genes.sd.dd,rhythmic.genes.sd.12)

genes.two.peaks.sd.one.peak.ll.dd <- intersect(genes.two.peaks.sd.one.peak.ll,
                                               genes.two.peaks.sd.one.peak.dd)

length(genes.two.peaks.sd.one.peak.ll.dd)
length(genes.two.peaks.sd.one.peak.ll.dd) / length(rhythmic.genes.sd.12)

complete.sd.dd.rhythmic.genes <- unique(c(rhythmic.genes.sd.dd,rhythmic.genes.sd.dd.12))
length(complete.sd.dd.rhythmic.genes)
length(complete.sd.dd.rhythmic.genes)/length(complete.sd.rhythmic.genes)
length(complete.sd.dd.rhythmic.genes)/number.genes

length(setdiff(complete.sd.rhythmic.genes,complete.sd.dd.rhythmic.genes))
non.rhythmic.sd.dd.genes <- setdiff(complete.sd.rhythmic.genes, complete.sd.dd.rhythmic.genes)
```

We found that 65.63% of the genes identified previously as rhythmic under SD conditions still presented cycling expression patterns when transferred to free running conditions consisting of constant darkness. Specifically, we identified 4006 circadian genes under SD conditions comprising 52.24% of the entire genome. The 2098 remaining genes cycling under SD conditions presented either a flat or noisy expression profile under DD conditions.   

```{r sd_rhytmic_ll_dd}
rhythmic.sd.ll.dd <- intersect(complete.sd.dd.rhythmic.genes,complete.sd.ll.rhythmic.genes)
length(rhythmic.sd.ll.dd)

length(rhythmic.sd.ll.dd)/number.genes

length(intersect(rhythmic.ld.ll.dd, rhythmic.sd.ll.dd))
length(intersect(rhythmic.ld.ll.dd, rhythmic.sd.ll.dd)) / number.genes

grid.newpage()
draw.pairwise.venn(area1 = length(complete.sd.ll.rhythmic.genes),
                   area2 = length(complete.sd.dd.rhythmic.genes),
                   cross.area = length(intersect(complete.sd.ll.rhythmic.genes,complete.sd.dd.rhythmic.genes)),
                   lwd = 6,alpha = 0.7,fill = c("lightred","red"),cex = 3,category = c("SD/LL","SD/DD"),cat.cex = 3,cat.pos = c(-10,10))

write.table(x = subset(gene.annotation, geneID %in% rhythmic.sd.ll.dd),file="gene_sets/SD_LL_DD_rhythmic_genes.tsv",row.names = F)

rhythmic.only.ll.or.dd <- setdiff(unique(c(complete.sd.dd.rhythmic.genes,complete.sd.ll.rhythmic.genes)),rhythmic.sd.ll.dd)
length(rhythmic.only.ll.or.dd)
length(rhythmic.only.ll.or.dd) / number.genes

write.table(x = subset(gene.annotation, geneID %in% rhythmic.only.ll.or.dd),file="gene_sets/SD_and_LL_or_DD_rhythmic_genes.tsv",row.names = F)

sd.rhythmic.only.ll <- setdiff(complete.sd.ll.rhythmic.genes,rhythmic.sd.ll.dd)
length(sd.rhythmic.only.ll)
write.table(x = sd.rhythmic.only.ll, file="gene_sets/SD_and_LL_rhythmic_genes.tsv",row.names = F)

sd.rhythmic.only.dd <- setdiff(complete.sd.dd.rhythmic.genes,rhythmic.sd.ll.dd)
length(sd.rhythmic.only.dd)
write.table(x = rhythmic.only.dd, file="gene_sets/SD_and_DD_rhythmic_genes.tsv",row.names = F)
```

Integrating the results from the two free running conditions analysed in this study, namely, constant light and constant dark we identified 1647 genes whose rhythmicity is independent from the alternating light/dark cycles after LD entrainment. In contrast, 2821 genes (36.79%) only maintained its rhythmicity after LD entrainment either under constant light or dark. Specifically, 1664 genes exhibiting only rhythmicity under constant dark and 1157 genes under constant light. These genes need either an light or dark input to maintain their rhythmicity loosing it under the free running conditions excluding the corresponding necessary input.

## **Identification of light/dark regulated genes after LD entrainment**

We now focus on the genes exhibiting rhythmicity exclusively under alternating light and dark cycles. We look for genes that are light activated and dark repressed or viceversa. We say a gene is light (or dark) repressed when under constant light (dark) its average expression level is less than its average expression level under alternating light dark cycles. Additionally, the average gene expression level during the first day needs to be greater or equal to the average gene expression level on the second day under free running conditions. Similarly, we define ligth (or dark) activated.  

```{r ld_dd_activated_repressed_by_darkness}
length(complete.ld.rhythmic.genes)
potential.light.dark.regulated <- setdiff(complete.ld.rhythmic.genes,unique(c(rhythmic.ld.ll.dd,rhythmic.only.dd,rhythmic.only.ll)))
length(potential.light.dark.regulated)

non.rhythmic.ld.dd.mean.expression.cycle <- apply(X = gene.expression[potential.light.dark.regulated,ld.zt.i],MARGIN = 1,FUN = mean)

non.rhythmic.ld.dd.mean.expression.dd <- apply(X = gene.expression[potential.light.dark.regulated,ld.dd.zt.i],MARGIN = 1,FUN = mean)


ld.potential.dark.repressed <- names(which(non.rhythmic.ld.dd.mean.expression.cycle > non.rhythmic.ld.dd.mean.expression.dd))
length(ld.potential.dark.repressed)
ld.potential.dark.repressed.expression.1 <- apply(X = gene.expression[ld.potential.dark.repressed,ld.dd.zt.i[1,]],MARGIN = 1,FUN = mean)
ld.potential.dark.repressed.expression.2 <- apply(X = gene.expression[ld.potential.dark.repressed,ld.dd.zt.i[2,]],MARGIN = 1,FUN = mean)
ld.dark.repressed <- names(which(ld.potential.dark.repressed.expression.1 >= ld.potential.dark.repressed.expression.2))
length(ld.dark.repressed)

ld.potential.dark.activated <- names(which(non.rhythmic.ld.dd.mean.expression.cycle < non.rhythmic.ld.dd.mean.expression.dd))
length(ld.potential.dark.activated)
ld.potential.dark.activated.expression.1 <- apply(X = gene.expression[ld.potential.dark.activated,ld.dd.zt.i[1,]],MARGIN = 1,FUN = mean)
ld.potential.dark.activated.expression.2 <- apply(X = gene.expression[ld.potential.dark.activated,ld.dd.zt.i[2,]],MARGIN = 1,FUN = mean)
ld.dark.activated <- names(which(ld.potential.dark.activated.expression.1 <= ld.potential.dark.activated.expression.2))
length(ld.dark.activated)

ld.no.clear.effect.light <- setdiff(potential.light.dark.regulated, 
                                   c(ld.dark.activated,ld.dark.repressed))
length(ld.no.clear.effect.dark)
```

```{r ld_ll_activated_repressed_by_light}
non.rhythmic.ld.ll.mean.expression.cycle <- apply(X = gene.expression[potential.light.dark.regulated,ld.zt.i],MARGIN = 1,FUN = mean)

non.rhythmic.ld.ll.mean.expression.ll <- apply(X = gene.expression[potential.light.dark.regulated,ld.ll.zt.i],MARGIN = 1,FUN = mean)


ld.potential.light.repressed <- names(which(non.rhythmic.ld.ll.mean.expression.cycle > non.rhythmic.ld.ll.mean.expression.ll))
length(ld.potential.light.repressed)
ld.potential.light.repressed.expression.1 <- apply(X = gene.expression[ld.potential.light.repressed,ld.ll.zt.i[1,]],MARGIN = 1,FUN = mean)
ld.potential.light.repressed.expression.2 <- apply(X = gene.expression[ld.potential.light.repressed,ld.ll.zt.i[2,]],MARGIN = 1,FUN = mean)
ld.light.repressed <- names(which(ld.potential.light.repressed.expression.1 >= ld.potential.light.repressed.expression.2))
length(ld.light.repressed)

ld.potential.light.activated <- names(which(non.rhythmic.ld.ll.mean.expression.cycle < non.rhythmic.ld.ll.mean.expression.ll))
length(ld.potential.light.activated)
ld.potential.light.activated.expression.1 <- apply(X = gene.expression[ld.potential.light.activated,ld.ll.zt.i[1,]],MARGIN = 1,FUN = mean)
ld.potential.light.activated.expression.2 <- apply(X = gene.expression[ld.potential.light.activated,ld.ll.zt.i[2,]],MARGIN = 1,FUN = mean)
ld.light.activated <- names(which(ld.potential.light.activated.expression.1 <= ld.potential.light.activated.expression.2))
length(ld.light.activated)

ld.no.clear.effect.light <- setdiff(potential.light.dark.regulated, 
                                   c(ld.light.activated,ld.light.repressed))
length(ld.no.clear.effect.light)
```


```{r ld_light_dark_activated_repressed_viceversa}
length(ld.light.activated)
length(ld.dark.repressed)

ld.light.activated.dark.repressed <- intersect(ld.light.activated,ld.dark.repressed)
length(ld.light.activated.dark.repressed)
length(ld.light.activated.dark.repressed) / number.genes


length(ld.light.repressed)
length(ld.dark.activated)

ld.light.repressed.dark.activated <- intersect(ld.light.repressed,ld.dark.activated)
length(ld.light.repressed.dark.activated)
length(ld.light.repressed.dark.activated) / number.genes

ld.light.dark.regulated <- c(ld.light.activated.dark.repressed,ld.light.repressed.dark.activated)
length(ld.light.dark.regulated)
length(ld.light.dark.regulated)/number.genes

length(potential.light.dark.regulated)
ld.no.clear.effect.light.dark.regulation <- setdiff(potential.light.dark.regulated,ld.light.dark.regulated)
length(ld.no.clear.effect.light.dark.regulation)

write.table(x = ld.light.activated.dark.repressed, file="gene_sets/LD_light_activated_dark_repressed.tsv",row.names = F)

write.table(x = ld.light.repressed.dark.activated, file="gene_sets/LD_light_repressed_dark_activated.tsv",row.names = F)

write.table(x = ld.no.clear.effect.light.dark.regulation, file="gene_sets/LD_no_clear_effect_light_dark_regulation.tsv",row.names = F)
```

We first identified 1733 genes that only present rhythmicity under long day conditions without presenting any rhythmicity under constant light or dark. Here 331 genes activated by light and repressed by dark and 56 activated by dark and repressed by light. The 1346 remaining genes did not present a clear regulatory effect by constant light or dark. 

## **Identification of light/dark regulated genes after SD entrainment**


```{r sd_dd_activated_repressed_by_darkness}
length(complete.sd.rhythmic.genes)
potential.light.dark.regulated <- setdiff(complete.sd.rhythmic.genes,unique(c(rhythmic.sd.ll.dd,rhythmic.only.dd,rhythmic.only.ll)))
length(potential.light.dark.regulated)

non.rhythmic.sd.dd.mean.expression.cycle <- apply(X = gene.expression[potential.light.dark.regulated,sd.zt.i],MARGIN = 1,FUN = mean)

non.rhythmic.sd.dd.mean.expression.dd <- apply(X = gene.expression[potential.light.dark.regulated,sd.dd.zt.i],MARGIN = 1,FUN = mean)


sd.potential.dark.repressed <- names(which(non.rhythmic.sd.dd.mean.expression.cycle > non.rhythmic.sd.dd.mean.expression.dd))
length(sd.potential.dark.repressed)
sd.potential.dark.repressed.expression.1 <- apply(X = gene.expression[sd.potential.dark.repressed,sd.dd.zt.i[1,]],MARGIN = 1,FUN = mean)
sd.potential.dark.repressed.expression.2 <- apply(X = gene.expression[sd.potential.dark.repressed,sd.dd.zt.i[2,]],MARGIN = 1,FUN = mean)
sd.dark.repressed <- names(which(sd.potential.dark.repressed.expression.1 >= sd.potential.dark.repressed.expression.2))
length(sd.dark.repressed)

sd.potential.dark.activated <- names(which(non.rhythmic.sd.dd.mean.expression.cycle < non.rhythmic.sd.dd.mean.expression.dd))
length(sd.potential.dark.activated)
sd.potential.dark.activated.expression.1 <- apply(X = gene.expression[sd.potential.dark.activated,sd.dd.zt.i[1,]],MARGIN = 1,FUN = mean)
sd.potential.dark.activated.expression.2 <- apply(X = gene.expression[sd.potential.dark.activated,sd.dd.zt.i[2,]],MARGIN = 1,FUN = mean)
sd.dark.activated <- names(which(sd.potential.dark.activated.expression.1 <= sd.potential.dark.activated.expression.2))
length(sd.dark.activated)

sd.no.clear.effect.dark <- setdiff(non.rhythmic.sd.dd.genes, 
                                   c(sd.dark.activated,sd.dark.repressed))
length(sd.no.clear.effect.dark)
```

We further analysis the expression parttern of the 2098 whose rhytmic expression pattern depend of the alternation of light and dark cycles identified under constant dark. We identified dark repressed/activated genes as those presenting a decreasing/increasing mean expression level over the two consecutive days under constant dark. Specifically, we found 1216 dark repressed genes and 288 dark activated genes. The remaining 594 genes did not present a clear activation or repression effect produced by constant dark after SD entrainment. 

```{r sd_ll_activated_repressed_by_light}
non.rhythmic.sd.ll.mean.expression.cycle <- apply(X = gene.expression[potential.light.dark.regulated,sd.zt.i],MARGIN = 1,FUN = mean)

non.rhythmic.sd.ll.mean.expression.ll <- apply(X = gene.expression[potential.light.dark.regulated,sd.ll.zt.i],MARGIN = 1,FUN = mean)


sd.potential.light.repressed <- names(which(non.rhythmic.sd.ll.mean.expression.cycle > non.rhythmic.sd.ll.mean.expression.ll))
length(sd.potential.light.repressed)
sd.potential.light.repressed.expression.1 <- apply(X = gene.expression[sd.potential.light.repressed,sd.ll.zt.i[1,]],MARGIN = 1,FUN = mean)
sd.potential.light.repressed.expression.2 <- apply(X = gene.expression[sd.potential.light.repressed,sd.ll.zt.i[2,]],MARGIN = 1,FUN = mean)
sd.light.repressed <- names(which(sd.potential.light.repressed.expression.1 >= sd.potential.light.repressed.expression.2))
length(sd.light.repressed)

sd.potential.light.activated <- names(which(non.rhythmic.sd.ll.mean.expression.cycle < non.rhythmic.sd.ll.mean.expression.ll))
length(sd.potential.light.activated)
sd.potential.light.activated.expression.1 <- apply(X = gene.expression[sd.potential.light.activated,sd.ll.zt.i[1,]],MARGIN = 1,FUN = mean)
sd.potential.light.activated.expression.2 <- apply(X = gene.expression[sd.potential.light.activated,sd.ll.zt.i[2,]],MARGIN = 1,FUN = mean)
sd.light.activated <- names(which(sd.potential.light.activated.expression.1 <= sd.potential.light.activated.expression.2))
length(sd.light.activated)

sd.no.clear.effect.light <- setdiff(non.rhythmic.sd.ll.genes, 
                                   c(sd.light.activated,sd.light.repressed))
length(sd.no.clear.effect.light)




```

We further analysis the expression parttern of the 4578 whose rhythmic expression pattern depend of the alternation of light and dark cycles identified under constant light. We identified light repressed/activated genes as those presenting a decreasing/increasing mean expression level over the two consecutive days under constant light. Specifically, we found 1741 light activated genes and 997 light repressed genes. The remaining 1840 genes did not present a clear activation or repression effect produced by constant light after SD entrainment. 

```{r sd_light_dark_activated_repressed_viceversa}
length(sd.light.activated)
length(sd.dark.repressed)

sd.light.activated.dark.repressed <- intersect(sd.light.activated,sd.dark.repressed)
length(sd.light.activated.dark.repressed)
length(sd.light.activated.dark.repressed) / number.genes

length(intersect(sd.light.activated.dark.repressed,ld.light.activated.dark.repressed))

length(sd.light.repressed)
length(sd.dark.activated)

sd.light.repressed.dark.activated <- intersect(sd.light.repressed,sd.dark.activated)
length(sd.light.repressed.dark.activated)
length(sd.light.repressed.dark.activated) / number.genes

sd.light.dark.regulated <- c(sd.light.activated.dark.repressed,sd.light.repressed.dark.activated)
length(sd.light.dark.regulated)
length(sd.light.dark.regulated)/number.genes

length(potential.light.dark.regulated)
sd.no.clear.effect.light.dark.regulation <- setdiff(potential.light.dark.regulated,sd.light.dark.regulated)
length(sd.no.clear.effect.light.dark.regulation)

write.table(x = sd.light.activated.dark.repressed, file="gene_sets/SD_light_activated_dark_repressed.tsv",row.names = F)

write.table(x = sd.light.repressed.dark.activated, file="gene_sets/SD_light_repressed_dark_activated.tsv",row.names = F)

write.table(x = sd.no.clear.effect.light.dark.regulation, file="gene_sets/SD_no_clear_effect_light_dark_regulation.tsv",row.names = F)
```

We first identified 1733 genes that only present rhythmicity under long day conditions without presenting any rhythmicity under constant light or dark. Here 331 genes activated by light and repressed by dark and 56 activated by dark and repressed by light. The 1346 remaining genes did not present a clear regulatory effect by constant light or dark. 



```{r}
data.sd.dd <- matrix(c(length(complete.sd.dd.rhythmic.genes),length(setdiff(complete.sd.rhythmic.genes,complete.sd.dd.rhythmic.genes)),length(non.rhythmic.sd.genes)),nrow=3)
sum(c(length(complete.sd.dd.rhythmic.genes),length(setdiff(complete.sd.rhythmic.genes,complete.sd.dd.rhythmic.genes)),length(non.rhythmic.sd.genes)))

barplot(data.sd.dd, lwd=2, names.arg = "",cex.names = 2,cex.axis = 1.5,
        col=c("red","lightpink","white"), 
        border="black", 
        font.axis=2, 
        xlab="",ylim=c(0,8000))

length(intersect(complete.ld.dd.rhythmic.genes,complete.sd.dd.rhythmic.genes))
length(setdiff(complete.ld.dd.rhythmic.genes,complete.sd.dd.rhythmic.genes))
length(setdiff(complete.sd.dd.rhythmic.genes,complete.ld.dd.rhythmic.genes))







complete.ld.sd.rhythmic.genes <- intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes)

grid.newpage()
draw.pairwise.venn(area1 = length(intersect(complete.ld.sd.rhythmic.genes,complete.ld.dd.rhythmic.genes)),area2 = length(intersect(complete.ld.sd.rhythmic.genes,complete.sd.dd.rhythmic.genes)),cross.area = length(intersect(complete.ld.sd.rhythmic.genes,intersect(complete.ld.dd.rhythmic.genes,complete.sd.dd.rhythmic.genes))),lwd = 3,category = c("LD","SD"),euler.d = T,col = c("blue","red"),fill = c(" blue","red"),alpha = 0.3,cex = 2,cat.cex = 2)


length(intersect(intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes),setdiff(complete.sd.ll.rhythmic.genes,complete.ld.ll.rhythmic.genes)))

length(intersect(intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes),setdiff(complete.ld.ll.rhythmic.genes,complete.sd.ll.rhythmic.genes)))


complete.ld.sd.rhythmic.genes <- intersect(complete.ld.rhythmic.genes,complete.sd.rhythmic.genes)
complete.ld.sd.ll.rhythmic.genes <- intersect(complete.ld.ll.rhythmic.genes,complete.sd.ll.rhythmic.genes)
complete.ld.sd.dd.rhythmic.genes <- intersect(complete.ld.dd.rhythmic.genes,complete.sd.dd.rhythmic.genes)

length(intersect(intersect(complete.ld.sd.rhythmic.genes,complete.ld.sd.ll.rhythmic.genes),
          complete.ld.sd.dd.rhythmic.genes))





rhythmic.sd.ll.dd <- intersect(complete.sd.dd.rhythmic.genes,complete.sd.ll.rhythmic.genes)
length(rhythmic.sd.ll.dd)

rhythmic.sd.ll <- setdiff(complete.sd.ll.rhythmic.genes,complete.sd.dd.rhythmic.genes)
length(rhythmic.sd.ll)

rhythmic.sd.dd <- setdiff(complete.sd.dd.rhythmic.genes,complete.sd.ll.rhythmic.genes)
length(rhythmic.sd.dd)

rhythmic.sd <- setdiff(non.rhythmic.sd.dd.genes, complete.sd.ll.rhythmic.genes)
length(rhythmic.sd)

length(non.rhythmic.sd.genes)
```


In order to explore the effect of the transition to continuous light on rhythmic gene expression patterns we compared three rhythmic characteristics of the expression profiles under LD or SD to the corresponding ones under LL. Specifically we compared amplitude or half of the difference between the peak and trough of the expression profile; mesor or mean expression level around which the expression profile oscillates and phase or time at which the expression profile peaks. This was carried out using the R package circacompare that performs a fitting to the data as cosinusoidal curve with a particular parametrization that is used to test for statistical significant differences between the two circadian patterns under comparison. 


Next we perform the analysis for circadian genes under SD conditions. It is important to notice that the results of circacompare sorts the samples in alphabetical order. Specifically, the LL conditions is first and SD condition is second. Therefore, to check for a decrease in amplitude in LL with respect to SD we have to check for positive values.

```{r sd_ll_amplitude_mesor_phase}
sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")

circacompare.sd.ll <- matrix(nrow=length(complete.sd.ll.rhythmic.genes),ncol=15)
rownames(circacompare.sd.ll) <- complete.sd.ll.rhythmic.genes

for(i in 1:length(complete.sd.ll.rhythmic.genes))
{
  gene.i <- complete.sd.ll.rhythmic.genes[i]
  
  sd.expression.i <- sd.ll.gene.expression[gene.i,c(paste(sd.zt,2,sep="_"),paste(sd.zt,3,sep="_"))]
  ll.expression.i <- sd.ll.gene.expression[gene.i,c(paste(sd.zt,4,sep="_"),paste(sd.zt,5,sep="_"))]
  
  time.points <- seq(from=0,by=4,length.out = 12)
  
  sd.ll.df <- data.frame(time=c(time.points,time.points),
                         measure=c(sd.expression.i, ll.expression.i),
                         group=c(rep("sd",12),rep("ll",12)))
  
  out.i <- circacompare(x = sd.ll.df, col_time = "time", col_group = "group", col_outcome = "measure",alpha_threshold = 1)
  circacompare.sd.ll[i,] <- out.i[[2]][,2]
}

colnames(circacompare.sd.ll) <- out.i[[2]][,1]

par(lwd=3)
boxplot(circacompare.sd.ll[,"sd amplitude estimate" ],circacompare.sd.ll[,"ll amplitude estimate" ],outline=F,col=c("red","lightsalmon"),names=c("SD","LL"),cex.axis=2)
sum(circacompare.sd.ll[,"Amplitude difference estimate" ] > 0)
sum(circacompare.sd.ll[,"Amplitude difference estimate" ] > 0) / length(complete.sd.ll.rhythmic.genes)
sum(circacompare.sd.ll[,"P-value for amplitude difference" ] < 0.05)
sum(circacompare.sd.ll[,"P-value for amplitude difference" ] < 0.05)/length(complete.sd.ll.rhythmic.genes)

boxplot(circacompare.sd.ll[,"sd mesor estimate" ],circacompare.sd.ll[,"ll mesor estimate" ],outline=F,col=c("red","lightsalmon"),names=c("SD","LL"),cex.axis=2)
sum(circacompare.sd.ll[,"Mesor difference estimate" ] < 0) /length(complete.sd.ll.rhythmic.genes)
sum(circacompare.sd.ll[,"Mesor difference estimate" ] > 0) /length(complete.sd.ll.rhythmic.genes)
sum(circacompare.sd.ll[,"P-value for mesor difference" ] < 0.05)/length(complete.sd.ll.rhythmic.genes)

sum(circacompare.sd.ll[,"P-value for difference in phase" ] < 0.05)
```
Similar to the results under LD conditons, we observed that most circadian genes under SD conditions, 93.11%, presented a sharp decrease in amplitude when transferred to LL, being significant in almost half of them with p-value < 0.05. Nevertheless, no clear effect was observed over the mesor or phase of gene expression profiles that remained in LL similar to the values in SD. 

The observed drastic reduction in amplitude together with the conservation in mesor and phase of the rhythmic gene expression profiles when transferred from LD or SD to LL conditions could be explained by two different scenarios. Under LD and SD conditions the entire cell culture is highly synchronous with most cells presenting coordinated rhythmic gene expression profiles. The transition to continuous light could produce the cell culture to become asynchronous. Individual cells would still be presenting an almost completely rhythmic transcriptome with a conservation in amplitude and mesor but out of phase with respect to the other cells in the culture. Since our data only capture the average transcriptomic state of the entire culture we observe a decrease in amplitude but a conservation in mesor and phase. In the other plausible scenario the entire cell culture would remain synchronous and the continuous light condition would produce an actual reduction in amplitude reflecting a deterioration in individual cell rhythmic transcriptomes. In a third possible scenario a combination of the two previous scenarios would take place. In order to discriminate between these possible scenarios for each individual gene single cell RNA-seq data would be required.    

The following instructions recreate the three possible scenarios and their effect over rhythmic gene expression patterns.

```{r three_scenarios}
wave.form <- function(mesor, amplitude,period,phase,time=seq(from=0,to=48,by=0.01))
{
  y <- mesor + amplitude*cos(0.0174533*period*(time - phase))
  return(y)
}


## effect on synchronization
plot(x=0,y=0,col="white",ylim=c(20,80),xlim=c(0,120),xlab="",ylab="",axes=F)
time <- seq(from=0,to=72,by=0.01)
time.2 <- seq(from=72,to=120,by=0.01)
N <- 50
norm.random.1 <- rnorm(n = N,mean = 0,sd = 1)
norm.random.2 <- rnorm(n = N,mean = 0,sd = 6)
syn.waves <- matrix(data = 0,nrow = N,ncol = length(time))
asyn.waves <- matrix(data = 0,nrow = N,ncol = length(time.2))

for(i in 1:N)
{
  syn.waves[i,] <- wave.form(mesor = 50,amplitude = 20,period = 15,phase = 16+norm.random.1[i],time=time)
  asyn.waves[i,] <- wave.form(mesor = 50,amplitude = 20,period = 15,phase = 16+norm.random.2[i],time=time.2)
  lines(time,syn.waves[i,],type="l",lwd=1,col="azure2")
  lines(time.2,asyn.waves[i,],type="l",lwd=1,col="azure2")
  
}

lines(x=c(time,time.2),y=c(colMeans(syn.waves),colMeans(asyn.waves)),type="l",lwd=5,col="blue")

polygon(x = c(0,16,16,0),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(16,24,24,16),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(24,40,40,24),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(40,48,48,40),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(48,64,64,48),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(64,72,72,64),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(72,88,88,72),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(88,96,96,88),y=c(25,25,22,22),lwd=4,border="blue",col="lightblue")
polygon(x = c(96,112,112,96),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(112,120,120,112),y=c(25,25,22,22),lwd=4,border="blue",col="lightblue")


## effect on amplitude
plot(x=0,y=0,col="white",ylim=c(20,80),xlim=c(0,120),xlab="",ylab="",axes=F)
time <- seq(from=0,to=72,by=0.01)
time.2 <- seq(from=72,to=120,by=0.01)
N <- 50
norm.random.1 <- - abs(rnorm(n = N,mean = 0,sd = 5))
norm.random.2 <- -abs(rnorm(n = N,mean = 0,sd = 15))
syn.waves <- matrix(data = 0,nrow = N,ncol = length(time))
asyn.waves <- matrix(data = 0,nrow = N,ncol = length(time.2))
for(i in 1:N)
{
  syn.waves[i,] <- wave.form(mesor = 50,amplitude = 20+norm.random.1[i],period = 15,phase = 16,time=time)
  asyn.waves[i,] <- wave.form(mesor = 50,amplitude = abs(15+norm.random.2[i]),period = 15,phase = 16,time=time.2)
  lines(time,syn.waves[i,],type="l",lwd=1,col="azure2")
  lines(time.2,asyn.waves[i,],type="l",lwd=1,col="azure2")
  
}

lines(x=c(time,time.2),y=c(colMeans(syn.waves),colMeans(asyn.waves)),type="l",lwd=5,col="blue")
polygon(x = c(0,16,16,0),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(16,24,24,16),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(24,40,40,24),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(40,48,48,40),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(48,64,64,48),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(64,72,72,64),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(72,88,88,72),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(88,96,96,88),y=c(25,25,22,22),lwd=4,border="blue",col="lightblue")
polygon(x = c(96,112,112,96),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(112,120,120,112),y=c(25,25,22,22),lwd=4,border="blue",col="lightblue")


## effect on synchronization + amplitude
plot(x=0,y=0,col="white",ylim=c(20,80),xlim=c(0,120),xlab="",ylab="",axes=F)
time <- seq(from=0,to=72,by=0.01)
time.2 <- seq(from=72,to=120,by=0.01)
N <- 50
norm.random.1 <- - abs(rnorm(n = N,mean = 0,sd = 1))
norm.random.2 <- -abs(rnorm(n = N,mean = 0,sd = 8))
norm.random.3 <- rnorm(n = N,mean = 0,sd = 1)
norm.random.4 <- rnorm(n = N,mean = 0,sd = 4)

syn.waves <- matrix(data = 0,nrow = N,ncol = length(time))
asyn.waves <- matrix(data = 0,nrow = N,ncol = length(time.2))
for(i in 1:N)
{
  syn.waves[i,] <- wave.form(mesor = 50,amplitude = 20+norm.random.1[i],period = 15+norm.random.3[i],phase = 16,time=time)
  asyn.waves[i,] <- wave.form(mesor = 50,amplitude = abs(20+norm.random.2[i]),period = 15+norm.random.4[i],phase = 16,time=time.2)
  lines(time,syn.waves[i,],type="l",lwd=1,col="azure2")
  lines(time.2,asyn.waves[i,],type="l",lwd=1,col="azure2")
  
}

lines(x=c(time,time.2),y=c(colMeans(syn.waves),colMeans(asyn.waves)),type="l",lwd=5,col="blue")
polygon(x = c(0,16,16,0),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(16,24,24,16),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(24,40,40,24),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(40,48,48,40),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(48,64,64,48),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(64,72,72,64),y=c(25,25,22,22),lwd=4,border="blue",col="blue")
polygon(x = c(72,88,88,72),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(88,96,96,88),y=c(25,25,22,22),lwd=4,border="blue",col="lightblue")
polygon(x = c(96,112,112,96),y=c(25,25,22,22),lwd=4,border="blue")
polygon(x = c(112,120,120,112),y=c(25,25,22,22),lwd=4,border="blue",col="lightblue")
```


## **Effect of double peaks under free running conditions**

```{r two_peaks}
library(circacompare)
sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")

circacompare.sd.ll.12 <- matrix(nrow=length(genes.two.peaks.sd.one.peak.ll.dd),ncol=15)
rownames(circacompare.sd.ll.12) <- genes.two.peaks.sd.one.peak.ll.dd
i <- 2
for(i in 1:length(complete.ld.ll.rhythmic.genes))
{
  gene.i <- genes.two.peaks.sd.one.peak.ll.dd[i]
  
  sd.ll.expression.i <- gene.expression[gene.i,c(paste(sd.zt,4,sep="_"),paste(ld.zt,5,sep="_"))]
  sd.dd.expression.i <- gene.expression[gene.i,c(paste(sd.zt,6,sep="_"),paste(ld.zt,7,sep="_"))]
  
  time.points <- seq(from=0,by=4,length.out = 12)
  
  sd.ll.dd.df <- data.frame(time=c(time.points,time.points),
                         measure=c(sd.ll.expression.i, sd.dd.expression.i),
                         group=c(rep("ll",12),rep("dd",12)))
  
  out.i <- circacompare(x = sd.ll.dd.df, col_time = "time", col_group = "group", col_outcome = "measure",alpha_threshold = 1)
  circacompare.ld.ll[i,] <- out.i[[2]][,2]
 }
```

## **Comparison between rhythmic gene sets under long day and short day photoperiods**


```{r ld_sd_circacompare}
grid.newpage()
draw.pairwise.venn(area1 = length(rhythmic.ld.ll.dd),
                   area2 = length(rhythmic.sd.ll.dd),
                   cross.area = length(intersect(rhythmic.ld.ll.dd,rhythmic.sd.ll.dd)),
                   lwd = 6,alpha = 0.7,fill = c("blue","red"),cex = 3,category = c("LD/LL/DD","SD/LL/DD"),cat.cex = 3,cat.pos = c(-10,10))


grid.newpage()
draw.pairwise.venn(area1 = length(ld.rhythmic.only.ll),
                   area2 = length(sd.rhythmic.only.ll),
                   cross.area = length(intersect(ld.rhythmic.only.ll,sd.rhythmic.only.ll)),
                   lwd = 6,alpha = 0.7,fill = c("blue","red"),cex = 3,category = c("LD/LL","SD/LL"),cat.cex = 3,cat.pos = c(-10,10))

grid.newpage()
draw.pairwise.venn(area1 = length(ld.rhythmic.only.dd),
                   area2 = length(sd.rhythmic.only.dd),
                   cross.area = length(intersect(ld.rhythmic.only.dd,sd.rhythmic.only.dd)),
                   lwd = 6,alpha = 0.7,fill = c("blue","red"),cex = 3,category = c("LD/DD","SD/DD"),cat.cex = 3,cat.pos = c(-10,10))


```



## **Comparison between rhythmic gene expression patterns under long day and short day photoperiods**

Next, we use the R package circacompare to study the global effect of the photoperiod over the amplitude, mesor and phase of rhythmic expression patterns. 

```{r ld_sd_circacompare}
ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")

circacompare.ld.sd <- matrix(nrow=length(complete.ld.sd.rhythmic.genes),ncol=15)
rownames(circacompare.ld.sd) <- complete.ld.sd.rhythmic.genes

for(i in 1:length(complete.ld.sd.rhythmic.genes))
{
  gene.i <- complete.ld.sd.rhythmic.genes[i]
  
  ld.expression.i <- gene.expression[gene.i,c(paste(ld.zt,1,sep="_"),paste(ld.zt,2,sep="_"),paste(ld.zt,3,sep="_"))]
  sd.expression.i <- gene.expression[gene.i,c(paste(sd.zt,1,sep="_"),paste(sd.zt,2,sep="_"),paste(sd.zt,3,sep="_"))]
  
  time.points <- seq(from=0,by=4,length.out = 18)
  
  ld.sd.df <- data.frame(time=c(time.points,time.points),
                         measure=c(ld.expression.i, sd.expression.i),
                         group=c(rep("ld",18),rep("sd",18)))
  
  out.i <- circacompare(x = ld.sd.df, col_time = "time", col_group = "group", col_outcome = "measure",alpha_threshold = 1)
  circacompare.ld.sd[i,] <- out.i[[2]][,2]
}

colnames(circacompare.ld.sd) <- out.i[[2]][,1]
```

```{r ld_sd_amplitude_effect}
sum(circacompare.ld.sd[,"Amplitude difference estimate" ] < 0)
sum(circacompare.ld.sd[,"Amplitude difference estimate" ] < 0) / length(complete.ld.sd.rhythmic.genes)
sum(circacompare.ld.sd[circacompare.ld.sd[,"Amplitude difference estimate" ] < 0,"P-value for amplitude difference" ] < 0.05)
genes.decrease.amplitude.effect.ld.sd <- names(which(circacompare.ld.sd[circacompare.ld.sd[,"Amplitude difference estimate" ] < 0,"P-value for amplitude difference" ] < 0.05)) 
sum(circacompare.ld.sd[circacompare.ld.sd[,"Amplitude difference estimate" ] < 0,"P-value for amplitude difference" ] < 0.05)/length(complete.ld.sd.rhythmic.genes)

sum(circacompare.ld.sd[,"Amplitude difference estimate" ] > 0)
sum(circacompare.ld.sd[,"Amplitude difference estimate" ] > 0) / length(complete.ld.sd.rhythmic.genes)

sum(circacompare.ld.sd[circacompare.ld.sd[,"Amplitude difference estimate" ] > 0,"P-value for amplitude difference" ] < 0.05)

sum(circacompare.ld.sd[circacompare.ld.sd[,"Amplitude difference estimate" ] > 0,"P-value for amplitude difference" ] < 0.05)/length(complete.ld.sd.rhythmic.genes)
```

First, we found that under short day conditions 2036 rhythmic genes corresponding to 46.12% exhibit a significant decrease in the amplitude of their rhythms. Whereas, only 123 corresponding to 2.33% significantly increase their amplitude in short day when compared to long day conditions.

```{r barplot_ld_sd_amplitude_effect}
par(lwd=3)
boxplot(circacompare.ld.sd[,"ld amplitude estimate" ],circacompare.ld.sd[,"sd amplitude estimate" ],outline=F,col=c("blue","red"),names=c("LD","SD"),cex.axis=2)
```

```{r ld_sd_mesor_effect}
sum(circacompare.ld.sd[,"Mesor difference estimate" ] < 0)
sum(circacompare.ld.sd[,"Mesor difference estimate" ] < 0) / length(complete.ld.sd.rhythmic.genes)
sum(circacompare.ld.sd[circacompare.ld.sd[,"Mesor difference estimate" ] < 0,"P-value for mesor difference" ] < 0.05)
sum(circacompare.ld.sd[circacompare.ld.sd[,"Mesor difference estimate" ] < 0,"P-value for mesor difference" ] < 0.05)/length(complete.ld.sd.rhythmic.genes)

sum(circacompare.ld.sd[,"Mesor difference estimate" ] > 0)
sum(circacompare.ld.sd[,"Mesor difference estimate" ] > 0) / length(complete.ld.sd.rhythmic.genes)

sum(circacompare.ld.sd[circacompare.ld.sd[,"Mesor difference estimate" ] > 0,"P-value for mesor difference" ] < 0.05)

sum(circacompare.ld.sd[circacompare.ld.sd[,"Mesor difference estimate" ] > 0,"P-value for mesor difference" ] < 0.05)/length(complete.ld.sd.rhythmic.genes)
```

No clear effect over the mesor was detected as 30.96% of the rhythmic genes significantly increase the mean value around which they oscillate and 19.48% significantly decrease it under short day conditions when compared to long day conditions.  

```{r barplot_ld_sd_mesor_effect}
par(lwd=3)
boxplot(circacompare.ld.sd[,"ld mesor estimate" ],circacompare.ld.sd[,"sd mesor estimate" ],outline=F,col=c("blue","red"),names=c("LD","SD"),cex.axis=2)
```

```{r ld_sd_phase_effect}
sum(circacompare.ld.sd[,"Phase difference estimate" ] < 0)
sum(circacompare.ld.sd[,"Phase difference estimate" ] < 0) / length(complete.ld.sd.rhythmic.genes)
sum(circacompare.ld.sd[circacompare.ld.sd[,"Phase difference estimate" ] < 0,"P-value for difference in phase" ] < 0.05)
genes.phase.effect.ld.sd <- names(which(circacompare.ld.sd[circacompare.ld.sd[,"Phase difference estimate" ] < 0,"P-value for difference in phase" ] < 0.05))
sum(circacompare.ld.sd[circacompare.ld.sd[,"Phase difference estimate" ] < 0,"P-value for difference in phase" ] < 0.05)/length(complete.ld.sd.rhythmic.genes)
```

The most evident effect in the rhythmic gene expression patterns observed when comparing
short day and long day conditions consists of a phase anticipation. Specifically, 3424 genes comprising 64.95% of the rhtymic genes exhibited a significantly anticipated phase under short day conditions when compared to long day conditions. The histograms below present the number of rhythmic genes with a specific phase peaking at the corresponding time point. Under long day conditions we observe a low number of rhythmic genes around 200 with phase during the light period from ZT0 to ZT12.       


```{r phase_histograms}
par(lwd=3)
hist(circacompare.ld.sd[,"ld peak time" ],col="blue",
     ylim=c(0,1200),xlim=c(0,24),main="",axes=FALSE,xlab="",ylab="")
axis(side = 2,lwd=3,cex.axis=1.3,las=2)
hist(circacompare.ld.sd[,"sd peak time" ],col="blue",
     ylim=c(0,1200),xlim=c(0,24),main="",axes=FALSE,xlab="",ylab="")
axis(side = 2,lwd=3,cex.axis=1.3,las=2)
```

The function below represents the expression profile of a specific gene under short and long day condtions.

```{r profile_function_ld_sd}
plot.ld.sd <- function(gene.id, gene.name, gene.expression)
{

  ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
  current.gene.expression.ld <- gene.expression[gene.id,c(paste(ld.zt,1,sep="_"),paste(ld.zt,2,sep="_"),paste(ld.zt,3,sep="_"))]
  
  sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
  current.gene.expression.sd <- gene.expression[gene.id,c(paste(sd.zt,1,sep="_"),paste(sd.zt,2,sep="_"),paste(sd.zt,3,sep="_"))]

  max.expr <- max(c(current.gene.expression.ld, current.gene.expression.sd))
  min.expr <- min(c(current.gene.expression.ld, current.gene.expression.sd))
  
  range.expr <- (max.expr - min.expr)
  
  plot(x = -10,y= -10,axes=F,xlab="",ylab="",
       ylim=c(min.expr-0.2*range.expr, max.expr+0.2*range.expr),xlim=c(0,72),
       main=paste(gene.id, gene.name,sep=" - "),cex.main=2)
  lines(x = seq(from=0,by=4,to=68),current.gene.expression.ld,type="o",lwd=5,col="blue")
  lines(x = seq(from=0,by=4,to=68),current.gene.expression.sd,type="o",lwd=5,col="red")
  axis(side=2,lwd=3,las=2,cex.axis=1.5)

  for(i in 0:2)
  {
    current.line <- (max.expr - min.expr)/20
    
    polygon(x = c(24*i, 24*i+16, 24*i+16, 24*i),
            y=c(min.expr-current.line, min.expr-current.line, 
                min.expr-2.5*(current.line), min.expr-2.5*(current.line)),
            lwd=3,border="blue")
    polygon(x = c(24*i+16,24*(i+1),24*(i+1),24*i+16),
            y=c(min.expr-current.line, min.expr-current.line, 
                min.expr-2.5*(current.line), min.expr-2.5*(current.line)),
            lwd=3,border="blue",col="blue")
    
    polygon(x = c(24*i, 24*i+8, 24*i+8, 24*i),
            y=c(min.expr-3*current.line, min.expr-3*current.line, 
                min.expr-4.5*(current.line), min.expr-4.5*(current.line)),
            lwd=3,border="red")
    polygon(x = c(24*i+8,24*(i+1),24*(i+1),24*i+8),
            y=c(min.expr-3*current.line, min.expr-3*current.line, 
                min.expr-4.5*(current.line), min.expr-4.5*(current.line)),
            lwd=3,border="red",col="red")
  }
  
  return(0)  
}
```

Some examples of genes exhibiting the typical phase anticipation and amplitude reduction under short day conditions when compared to long day conditions. 

```{r profile_function_ld_sd}
plot.ld.sd(gene.id="ostta01g06150", gene.name="CYCB", gene.expression)
plot.ld.sd(gene.id="ostta01g00790", gene.name="ADS1", gene.expression)
```

```{r profile_function, eval= FALSE}
plot.ld.nd.sd <- function(gene.id, gene.name, ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=T,scale=T)
{
  if(nd)
  {
    scale <- T
  }
  
  current.gene.expression.ld <- 0
  current.gene.expression.nd <- 0
  current.gene.expression.sd <- 0

  if(ld)
  {
    ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
    current.gene.expression.ld <- ld.gene.expression[gene.id,c(paste(ld.zt,1,sep="_"),paste(ld.zt,2,sep="_"),paste(ld.zt,3,sep="_"))]
    if(scale)
    {
      current.gene.expression.ld <- scale(current.gene.expression.ld)
    } 
  }

  if(sd)
  {
    sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
    current.gene.expression.sd <-        sd.gene.expression[gene.id,c(paste(sd.zt,1,sep="_"),paste(sd.zt,2,sep="_"),paste(sd.zt,3,sep="_"))]
    if(scale)
    {
      current.gene.expression.sd <- scale(current.gene.expression.sd)
    }
}

  if(nd)
  {
  nd.zt <- paste0("ZT",seq(from=0,to=21,by=3))
  current.gene.expression.nd <- scale(nd.gene.expression[gene.id,c(paste(nd.zt,1,sep="_"),paste(nd.zt,2,sep="_"),paste(nd.zt,3,sep="_"))])
}

  max.expr <- max(c(current.gene.expression.ld, current.gene.expression.nd, current.gene.expression.sd))
  min.expr <- min(c(current.gene.expression.ld, current.gene.expression.nd, current.gene.expression.sd))
  
  plot(x = -10,y= -10,axes=F,xlab="",ylab="",
       ylim=c(min.expr-2, max.expr),xlim=c(0,72),
       main=paste(gene.id, gene.name,sep=" - "),cex.main=2)
  
  if(!scale)
  {
    axis(side=2,lwd=3)
  }

  if(ld)
  {
  lines(x = seq(from=0,by=4,to=68),current.gene.expression.ld,type="o",lwd=3,col="blue")
}

  if(sd)
  {
  lines(x = seq(from=0,by=4,to=68),current.gene.expression.sd,type="o",lwd=3,col="red")
}

  if(nd)
  {
  lines(x = seq(from=0,by=3,to=69),current.gene.expression.nd,type="o",lwd=3,col="black")
}

  for(i in 0:2)
  {
  current.line <- 0.5
  
  if(ld)
  {
    polygon(x = c(24*i, 24*i+16, 24*i+16, 24*i),
            y=c(min.expr-current.line, min.expr-current.line, min.expr-(current.line+0.25), min.expr-(current.line +0.25)),
            lwd=2,border="blue")
    polygon(x = c(24*i+16,24*(i+1),24*(i+1),24*i+16),
            y=c(min.expr-current.line, min.expr-current.line, min.expr-(current.line+0.25), min.expr-(current.line +0.25)),
            lwd=2,border="blue",col="blue")
    current.line <- current.line + 0.5
  }
  
  if(nd)
  {
    polygon(x = c(24*i, 24*i+12, 24*i+12, 24*i),
            y=c(min.expr-current.line, min.expr-current.line, min.expr-(current.line+0.25), min.expr-(current.line +0.25)),
            lwd=2,border="black")
    polygon(x = c(24*i+12,24*(i+1),24*(i+1),24*i+12),
            y=c(min.expr-current.line, min.expr-current.line, min.expr-(current.line+0.25), min.expr-(current.line +0.25)),
            lwd=2,border="black",col="black")
    current.line <- current.line + 0.5
  }

  if(sd)
  {
    polygon(x = c(24*i, 24*i+8, 24*i+8, 24*i),
            y=c(min.expr-current.line, min.expr-current.line, min.expr-(current.line+0.25), min.expr-(current.line +0.25)),
            lwd=2,border="red")
    polygon(x = c(24*i+8,24*(i+1),24*(i+1),24*i+8),
            y=c(min.expr-current.line, min.expr-current.line, min.expr-(current.line+0.25), min.expr-(current.line +0.25)),
            lwd=2,border="red",col="red")
  }
}

  return(0)  
}

plot.ld.nd.sd(gene.id = "ostta01g02580",gene.name = "MCM6", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=F,nd=F,scale=F)

plot.ld.nd.sd(gene.id = "ostta04g00450",gene.name = "MCM5", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=T)

plot.ld.nd.sd(gene.id = "ostta01g01370",gene.name = "???", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=F)


# i <- 1
# plot.ld.nd.sd(gene.id = common.12.nd.sd[i],gene.name = "???", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=F,sd=T,nd=T)
# i <- i + 1
# 
# i <- i - 1
# plot.ld.nd.sd(gene.id = common.12.nd.sd[i],gene.name = "???", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=F)

i <- 1
plot.ld.nd.sd(gene.id = sd.specific.rhythmic.genes[i],gene.name = "???", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=F)
i <- i + 1
i <- i - 1


plot.ld.nd.sd(gene.id = "ostta13g01820",gene.name = "???", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=F)



plot.ld.ll <- function(gene.id, gene.name, gene.expression)
{
    ld.zt <- paste("ld",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
    current.gene.expression.ld.ll <- gene.expression[gene.id,c(paste(ld.zt,1,sep="_"),paste(ld.zt,2,sep="_"),paste(ld.zt,3,sep="_"),paste(ld.zt,4,sep="_"),paste(ld.zt,5,sep="_"))]

    min.expression <- min(current.gene.expression.ld.ll)
    max.expression <- max(current.gene.expression.ld.ll)
    range.expression <- max.expression - min.expression
    
    expression.step <- floor(range.expression / 5)
    
    plot(current.gene.expression.ld.ll,type="o",lwd=3,col="blue",axes=F,xlab="",ylab="FPKM",
         ylim=c(min.expression-expression.step,max.expression),
         cex.lab=1.3,main=paste(gene.id, gene.name,sep=" - "),cex.main=2)
    axis(side=2,lwd=3)
    axis(side = 1,pos = min.expression - 1.1*expression.step, at = seq(from=1,to=30),
         labels = rep(paste("ZT",seq(from=0,to=20,by=4)),5),las=2,lwd=3)
    
    polygon(x = c(1,5,5,1),y=c(min.expression-expression.step/2,
                               min.expression-expression.step/2,
                               min.expression-expression.step,
                               min.expression-expression.step),lwd=2,border="blue")
    
    polygon(x = c(5,7,7,5),y=c(min.expression-expression.step/2,
                               min.expression-expression.step/2,
                               min.expression-expression.step,
                               min.expression-expression.step),lwd=2,border="blue",col="blue")
    
    polygon(x = c(7,11,11,7),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="blue")
    
    polygon(x = c(11,13,13,11),y=c(min.expression-expression.step/2,
                                   min.expression-expression.step/2,
                                   min.expression-expression.step,
                                   min.expression-expression.step),lwd=2,border="blue",col="blue")
    
    polygon(x = c(13,17,17,13),y=c(min.expression-expression.step/2,
                                   min.expression-expression.step/2,
                                   min.expression-expression.step,
                                   min.expression-expression.step),lwd=2,border="blue")
    
    polygon(x = c(17,19,19,17),y=c(min.expression-expression.step/2,
                                   min.expression-expression.step/2,
                                   min.expression-expression.step,
                                   min.expression-expression.step),lwd=2,border="blue",col="blue")
    
    polygon(x = c(19,23,23,19),y=c(min.expression-expression.step/2,
                                   min.expression-expression.step/2,
                                   min.expression-expression.step,
                                   min.expression-expression.step),lwd=2,border="blue")
    
    polygon(x = c(23,25,25,23),y=c(min.expression-expression.step/2,
                                   min.expression-expression.step/2,
                                   min.expression-expression.step,
                                   min.expression-expression.step),lwd=2,border="blue",col="lightblue")
    
    polygon(x = c(25,29,29,25),y=c(min.expression-expression.step/2,
                                   min.expression-expression.step/2,
                                   min.expression-expression.step,
                                   min.expression-expression.step),lwd=2,border="blue")
    polygon(x = c(29,30,30,29),y=c(min.expression-expression.step/2,
                                   min.expression-expression.step/2,
                                   min.expression-expression.step,
                                   min.expression-expression.step),lwd=2,border="blue",col="lightblue")
    
}


i <- 1
plot.ld.ll(gene.id = "ostta04g00450",gene.name = "MCM5", gene.expression)


i <- 1
plot.ld.ll(gene.id = complete.ld.ll.rhythmic.genes[i],gene.name = i, gene.expression)
i <- i + 1


plot.sd.ll <- function(gene.id, gene.name, gene.expression)#, mean.expression.ld, mean.expression.sd)
{
    sd.zt <- paste("sd",paste0("zt",sprintf(fmt = "%02d",seq(from=0,to=20,by=4))),sep="_")
    current.gene.expression.sd.ll <- gene.expression[gene.id,c(paste(sd.zt,1,sep="_"),paste(sd.zt,2,sep="_"),paste(sd.zt,3,sep="_"),paste(sd.zt,4,sep="_"),paste(sd.zt,5,sep="_"))]
  
  min.expression <- min(current.gene.expression.sd.ll)
  max.expression <- max(current.gene.expression.sd.ll)
  range.expression <- max.expression - min.expression
  
  expression.step <- floor(range.expression / 5)
  
  plot(current.gene.expression.sd.ll,type="o",lwd=3,col="red",axes=F,xlab="",ylab="FPKM",
       ylim=c(min.expression-expression.step,max.expression),
       cex.lab=1.3,main=paste(gene.id, gene.name,sep=" - "),cex.main=2)
  axis(side=2,lwd=3)
  axis(side = 1,pos = min.expression - 1.1*expression.step, at = seq(from=1,to=30),
       labels = rep(paste("ZT",seq(from=0,to=20,by=4)),5),las=2,lwd=3)
  
  polygon(x = c(1,3,3,1),y=c(min.expression-expression.step/2,
                             min.expression-expression.step/2,
                             min.expression-expression.step,
                             min.expression-expression.step),lwd=2,border="red")
  
  polygon(x = c(3,7,7,3),y=c(min.expression-expression.step/2,
                             min.expression-expression.step/2,
                             min.expression-expression.step,
                             min.expression-expression.step),lwd=2,border="red",col="red")
  
  polygon(x = c(7,9,9,7),y=c(min.expression-expression.step/2,
                               min.expression-expression.step/2,
                               min.expression-expression.step,
                               min.expression-expression.step),lwd=2,border="red")
  
  polygon(x = c(9,13,13,9),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="red",col="red")
  
  polygon(x = c(13,15,15,13),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="red")
  
  polygon(x = c(15,19,19,15),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="red",col="red")
  
  polygon(x = c(19,21,21,19),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="red")
  
  polygon(x = c(21,25,25,21),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="red",col="lightsalmon")
  
  polygon(x = c(25,27,27,25),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="red")
  polygon(x = c(27,30,30,27),y=c(min.expression-expression.step/2,
                                 min.expression-expression.step/2,
                                 min.expression-expression.step,
                                 min.expression-expression.step),lwd=2,border="red",col="lightsalmon")
  
}


i <- 1
plot.sd.ll(gene.id = big.amplitude[i],gene.name = i, gene.expression)
i <- i + 1


sd.non.rhythmic.ll <- setdiff(complete.sd.rhythmic.genes,complete.sd.ll.rhythmic.genes)
plot.sd.ll(gene.id = sd.non.rhythmic.ll[i],gene.name = i, gene.expression)
i <- i + 1


plot.sd.ll(gene.id = rhythmic.genes.sd.12[i],gene.name = i, gene.expression)
i <- i + 1

i <- 1
plot.ld.ll(gene.id = complete.ld.ll.rhythmic.genes[i],gene.name = "Nitrogen regulatory PII-like, alpha/beta", gene.expression)
i <- i + 1
i <- i - 1

plot.ld.nd.sd(gene.id = "ostta01g05020",gene.name = "glnA", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=F)

plot.ld.nd.sd(gene.id = "ostta14g01900",gene.name = "gltS", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=F)

plot.ld.nd.sd(gene.id = "ostta01g05650",gene.name = "ATG8", ld.gene.expression,nd.gene.expression,sd.gene.expression,ld=T,sd=T,nd=F)


```